<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    
    <title>A Collection of Already Published Phylostratigraphic Maps and Divergence Maps | 拾荒者</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
    <meta name="description" content="General OverviewGene Age Inference is the foundation of most Evolutionary Transcriptomics studies.The concept behind Evolutionary Transcriptomics is to combine these gene age estimates with gene expre">
<meta property="og:type" content="article">
<meta property="og:title" content="A Collection of Already Published Phylostratigraphic Maps and Divergence Maps">
<meta property="og:url" content="http://blog.xieguigang.me/2016/07/18/published_phylomaps/index.html">
<meta property="og:site_name" content="拾荒者">
<meta property="og:description" content="General OverviewGene Age Inference is the foundation of most Evolutionary Transcriptomics studies.The concept behind Evolutionary Transcriptomics is to combine these gene age estimates with gene expre">
<meta property="og:image" content="https://raw.githubusercontent.com/xieguigang/xieguigang.github.io-hexo/master/images/qrcode/published_phylomaps.png">
<meta property="og:updated_time" content="2016-08-17T13:47:32.394Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="A Collection of Already Published Phylostratigraphic Maps and Divergence Maps">
<meta name="twitter:description" content="General OverviewGene Age Inference is the foundation of most Evolutionary Transcriptomics studies.The concept behind Evolutionary Transcriptomics is to combine these gene age estimates with gene expre">
<meta name="twitter:image" content="https://raw.githubusercontent.com/xieguigang/xieguigang.github.io-hexo/master/images/qrcode/published_phylomaps.png">
    

    
        <link rel="alternate" href="/" title="拾荒者" type="application/atom+xml" />
    

    
        <link rel="icon" href="http://xieguigang.me/favicon.ico" />
    

    <link rel="stylesheet" href="/vendor/font-awesome/css/font-awesome.min.css">
    <link rel="stylesheet" href="/vendor/open-sans/styles.css">
    <link rel="stylesheet" href="/vendor/source-code-pro/styles.css">

    <link rel="stylesheet" href="/css/style.css">

    <script src="/vendor/jquery/2.1.3/jquery.min.js"></script>
    
    
        <link rel="stylesheet" href="/vendor/fancybox/jquery.fancybox.css">
    
    
    
    

</head>

<body>
    <div id="container">
        <header id="header">
    <div id="header-main" class="header-inner">
        <div class="outer">
            <a href="/" id="logo">
                <i class="logo"></i>
                <span class="site-title">拾荒者</span>
            </a>
            <nav id="main-nav">
                
                    <a class="main-nav-link" href="/.">Home</a>
                
                    <a class="main-nav-link" href="/archives">Archives</a>
                
                    <a class="main-nav-link" href="/nuget">nuget</a>
                
                    <a class="main-nav-link" href="/about">About</a>
                
            </nav>
            
                
                <nav id="sub-nav">
                    <div class="profile" id="profile-nav">
                        <a id="profile-anchor" href="javascript:;">
                            <img class="avatar" src="https://raw.githubusercontent.com/xieguigang/xieguigang.github.io-hexo/master/images/my/WeChatImage636059681622385094.jpg" />
                            <i class="fa fa-caret-down"></i>
                        </a>
                    </div>
                </nav>
            
            <div id="search-form-wrap">

    <form class="search-form">
        <input type="text" class="ins-search-input search-form-input" placeholder="Search" />
        <button type="submit" class="search-form-submit"></button>
    </form>
    <div class="ins-search">
    <div class="ins-search-mask"></div>
    <div class="ins-search-container">
        <div class="ins-input-wrapper">
            <input type="text" class="ins-search-input" placeholder="Type something..." />
            <span class="ins-close ins-selectable"><i class="fa fa-times-circle"></i></span>
        </div>
        <div class="ins-section-wrapper">
            <div class="ins-section-container"></div>
        </div>
    </div>
</div>
<script>
(function (window) {
    var INSIGHT_CONFIG = {
        TRANSLATION: {
            POSTS: 'Posts',
            PAGES: 'Pages',
            CATEGORIES: 'Categories',
            TAGS: 'Tags',
            UNTITLED: '(Untitled)',
        },
        ROOT_URL: '/',
        CONTENT_URL: '/content.json',
    };
    window.INSIGHT_CONFIG = INSIGHT_CONFIG;
})(window);
</script>
<script src="/js/insight.js"></script>

</div>
        </div>
    </div>
    <div id="main-nav-mobile" class="header-sub header-inner">
        <table class="menu outer">
            <tr>
                
                    <td><a class="main-nav-link" href="/.">Home</a></td>
                
                    <td><a class="main-nav-link" href="/archives">Archives</a></td>
                
                    <td><a class="main-nav-link" href="/nuget">nuget</a></td>
                
                    <td><a class="main-nav-link" href="/about">About</a></td>
                
                <td>
                    
    <div class="search-form">
        <input type="text" class="ins-search-input search-form-input" placeholder="Search" />
    </div>

                </td>
            </tr>
        </table>
    </div>
</header>

        <div class="outer">
            
                

<aside id="profile">
    <div class="inner profile-inner">
        <div class="base-info profile-block">
            <img id="avatar" src="https://raw.githubusercontent.com/xieguigang/xieguigang.github.io-hexo/master/images/my/WeChatImage636059681622385094.jpg" />
            <h2 id="name">xieguigang</h2>
            <h3 id="title">GCModeller chief developer</h3>
            <span id="location"><i class="fa fa-map-marker"></i>GuiLin, China (中國 · 桂林)</span>
            <a id="follow" target="_blank" href="https://github.com/xieguigang">FOLLOW</a>
        </div>
        <div class="article-info profile-block">
            <div class="article-info-block">
                47
                <span>posts</span>
            </div>
            <div class="article-info-block">
                96
                <span>tags</span>
            </div>
        </div>
        
        <div class="profile-block social-links">
            <table>
                <tr>
                    
                    
                    <td>
                        <a href="http://github.com/xieguigang" target="_blank" title="github" class=tooltip>
                            <i class="fa fa-github"></i>
                        </a>
                    </td>
                    
                    <td>
                        <a href="/" target="_blank" title="twitter" class=tooltip>
                            <i class="fa fa-twitter"></i>
                        </a>
                    </td>
                    
                    <td>
                        <a href="/" target="_blank" title="rss" class=tooltip>
                            <i class="fa fa-rss"></i>
                        </a>
                    </td>
                    
                </tr>
            </table>
        </div>
        
    </div>
</aside>

            
            <section id="main"><article id="post-published_phylomaps" class="article article-type-post" itemscope itemprop="blogPost">
    <div class="article-inner">
        
        
        
            <header class="article-header">
                
    
        <h1 class="article-title" itemprop="name">
            A Collection of Already Published Phylostratigraphic Maps and Divergence Maps
        </h1>
    

                
                    <div class="article-meta">
                        
    <div class="article-date">
        <i class="fa fa-calendar"></i>
        <a href="/2016/07/18/published_phylomaps/">
            <time datetime="2016-07-17T16:00:00.000Z" itemprop="datePublished">2016-07-18</time>
        </a>
    </div>


                        
                        
    <div class="article-tag">
        <i class="fa fa-tag"></i>
        <a class="tag-link" href="/tags/Phylostratigraphic/">Phylostratigraphic</a>, <a class="tag-link" href="/tags/R/">R</a>
    </div>

                    </div>
                
            </header>
        
        <div class="article-entry" itemprop="articleBody">
        
            
            <h1 id="General-Overview"><a href="#General-Overview" class="headerlink" title="General Overview"></a>General Overview</h1><p>Gene Age Inference is the foundation of most <a href="https://github.com/HajkD/myTAI#perform-evolutionary-transcriptomics-with-r" target="_blank" rel="external">Evolutionary Transcriptomics</a> studies.<br>The concept behind Evolutionary Transcriptomics is to combine these gene age estimates with gene expression<br>data to quantify the average transcriptome age within a biological process of interest (<a href="http://biorxiv.org/content/early/2016/05/03/051565.abstract" target="_blank" rel="external">Drost et al., 2016, bioRxiv</a>).</p>
<p>In particular, this approach allowed the quantification of transcriptome conservation of animal and plant embryos passing through embryogenesis by first individually estimating the gene ages of specific animal and plant genomes and combining these gene age estimates with transcriptome data covering several stages of embryo development (<a href="http://www.nature.com/nature/journal/v468/n7325/full/nature09632.html" target="_blank" rel="external">Domazet-Loso and Tautz, 2010 <em>Nature</em></a> ; <a href="http://www.nature.com/nature/journal/v490/n7418/full/nature11394.html" target="_blank" rel="external">Quint, Drost et al., 2012 <em>Nature</em></a> ; <a href="http://mbe.oxfordjournals.org/content/32/5/1221" target="_blank" rel="external">Drost et al., 2015 <em>Mol. Biol. Evol.</em></a> ; <a href="http://mbe.oxfordjournals.org/content/early/2016/02/23/molbev.msw039.short?rss=1" target="_blank" rel="external">Drost et al., 2016 <em>Mol. Biol. Evol.</em></a>).<br><a id="more"></a></p>
<p>However, as intensely discussed in the past years (<a href="http://www.sciencedirect.com/science/article/pii/S016895251300111X" target="_blank" rel="external">Capra et al., 2013</a>; <a href="http://www.nature.com/nmeth/journal/v13/n5/pdf/nmeth.3830.pdf" target="_blank" rel="external">Altenhoff et al., 2016</a>; <a href="http://gbe.oxfordjournals.org/content/early/2016/06/03/gbe.evw113" target="_blank" rel="external">Liebeskind et al., 2016</a>), gene age inference is not a trivial task and might be biased in some currently existing approaches (<a href="http://gbe.oxfordjournals.org/content/early/2016/06/03/gbe.evw113" target="_blank" rel="external">Liebeskind et al., 2016</a>).</p>
<p>In particular, Moyers &amp; Zhang argue that <a href="http://www.sciencedirect.com/science/article/pii/S0168952507002995" target="_blank" rel="external">genomic phylostratigraphy</a> (a prominent BLAST based gene age inference method) 1) underestimates gene age for a considerable fraction of genes, 2) is biased for rapidly evolving proteins which are short, and/or their most conserved block of sites is small, and 3) these biases create spurious nonuniform distributions of various gene properties among age groups, many of which cannot be predicted a priori (<a href="http://mbe.oxfordjournals.org/content/32/1/258.long" target="_blank" rel="external">Moyers &amp; Zhang, 2015</a>;  <a href="http://mbe.oxfordjournals.org/content/33/5/1245" target="_blank" rel="external">Moyers &amp; Zhang, 2016</a>; <a href="http://gbe.oxfordjournals.org/content/early/2016/06/03/gbe.evw113" target="_blank" rel="external">Liebeskind et al., 2016</a>). However, these arguments are based on simulated data (and the approach has been questioned in favour for genomic phylostratigraphy; <a href="http://www.biorxiv.org/content/early/2016/06/26/060756.abstract" target="_blank" rel="external">Domazet-Loso et al., 2016</a>) and therefore an objective benchmarking set representing the tree of life is still missing.</p>
<p>Based on this debate a recent study suggested to perform gene age inference by combining thirteen common orthology inference algorithms to create gene age datasets and then characterize the error around each age-call on a per-gene and per-algorithm basis. Using this approach systematic error was found to be a large factor in estimating gene age, suggesting that simple consensus algorithms are not enough to give a reliable point estimate (<a href="http://gbe.oxfordjournals.org/content/early/2016/06/03/gbe.evw113" target="_blank" rel="external">Liebeskind et al., 2016</a>). However, by generating a consensus gene age and quantifying the possible error in each workflow step, <a href="http://gbe.oxfordjournals.org/content/early/2016/06/03/gbe.evw113" target="_blank" rel="external">Liebeskind et al., 2016</a> provide a very useful <a href="http://geneages.org/" target="_blank" rel="external">database</a> of<br>consensus gene ages for a variety of genomes. </p>
<p>Alternatively, <a href="https://bib.oxfordjournals.org/content/early/2016/04/20/bib.bbw034.full" target="_blank" rel="external">Stephen Smith, 2016</a> argues that <em>de novo</em> gene birth/death and gene family expansion/contraction studies should avoid drawing direct inferences of evolutionary relatedness from measures of sequence similarity alone, and should instead, where possible, use more rigorous phylogeny-based methods. For this purpose, I recommend researchers to consult the <a href="http://orthology.phylomedb.org/" target="_blank" rel="external">phylomedb database</a> to retrieve phylogeny-based gene orthology relationships and use these age estimates in combination with <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a>.</p>
<p>The <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a> package aims to provide a standard tool for Evolutionary Transcriptomics studies and relies<br>on gene age estimate tables as input. Hence, I recommend to follow the active discussion on gene age inference and to consult all available resources to robustly estimate gene age (for ex. use the consensus gene age estimates provided by<br><a href="http://gbe.oxfordjournals.org/content/early/2016/06/03/gbe.evw113" target="_blank" rel="external">Liebeskind et al., 2016</a> and <em>phylostratigraphic maps</em> generated by phylostratigraphy to quantify transcriptome age with <code>myTAI</code>).</p>
<p>In case researchers would like to perform <em>genomic phylostratigraphy</em>, a new tool named <a href="http://bioinformatics.oxfordjournals.org/content/early/2016/03/26/bioinformatics.btw122" target="_blank" rel="external">ORFanFinder</a> has recently been released and can be used to generate high quality phylostratigraphic maps.</p>
<p>Evidently, these advancements in gene age research are very recent and gene age inference is a very young and active field of genomic research. Therefore, many more studies need to address the robust and realistic inference<br>of gene age and a community standard is still missing.</p>
<p>To provide a comprehensive resource of gene ages, I accumulated all phylostratigraphic maps or sequence divergence maps that have been published to date and they might nevertheless be useful to study global patterns of transcriptome conservation in biological processes.</p>
<p>However, future Evolutionary Transcriptomics studies (this will include my own research) should consider these new advancements in gene age inference methods and concepts, because they will allow researchers to quantify transcriptome conservation or estimate the average transcriptome age with <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a> more accurately and more robustly. </p>
<p><strong>If your study recently published a phylostratigraphic map or sequence divergence map, please contact me (<a href="https://github.com/HajkD/published_phylomaps/issues" target="_blank" rel="external">https://github.com/HajkD/published_phylomaps/issues</a>) and I will gladly reference you and include your study in the following list.</strong></p>
<h2 id="A-Collection-of-Published-Phylostratigraphic-Maps-and-Divergence-Maps"><a href="#A-Collection-of-Published-Phylostratigraphic-Maps-and-Divergence-Maps" class="headerlink" title="A Collection of Published Phylostratigraphic Maps and Divergence Maps"></a>A Collection of Published Phylostratigraphic Maps and Divergence Maps</h2><p>The following studies include published <code>Phylostratigraphic Maps</code> and <code>Divergence Maps</code> based on similar approaches, but using different parameter sets.<br>This collection aims to store all published maps and furthermore, provides scripts for easy data retrieval to be able to integrate corresponding maps into a phylostranscriptomics workflow with <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a>.</p>
<p>The <a href="https://github.com/HajkD/myTAI/blob/master/vignettes/Introduction.Rmd" target="_blank" rel="external">Introduction to Phylotranscriptomics</a> Vignette introduces the integration of the following <code>Phylostratigraphic Maps</code> and <code>Divergence Maps</code> to perform custom phylotranscriptomic analyses.</p>
<h2 id="Tomislav-Domazet-Loso-Josip-Brajkovic-Diethard-Tautz-2007"><a href="#Tomislav-Domazet-Loso-Josip-Brajkovic-Diethard-Tautz-2007" class="headerlink" title="Tomislav Domazet-Lošo, Josip Brajković, Diethard Tautz, 2007"></a><a href="http://www.sciencedirect.com/science/article/pii/S0168952507002995" target="_blank" rel="external">Tomislav Domazet-Lošo, Josip Brajković, Diethard Tautz, 2007</a></h2><p><strong>Title</strong>: <em>A phylostratigraphy approach to uncover the genomic history of major adaptations in metazoan lineages</em></p>
<p>This study introduced <code>Phylostratigraphy</code> as computational method to quantify gene age in terms of sequence homology resulting in an organism specific <code>Phylostratigraphic Map</code>.</p>
<p>Published <code>Phylostratigraphic Map</code>:</p>
<ul>
<li><strong>Organisms</strong>: <em>Drosophila melanogaster</em> (fly)</li>
<li><strong>E-value cutoff</strong>: 1E-3 (blastp; protein sequences) and 1E-15 (tblastn; ESTs) </li>
<li><strong>Sequence type</strong>: Protein Sequences and ESTs</li>
<li><strong>Reference data bases</strong>: NCBI nr (protein); trace and EST archives (EST)</li>
</ul>
<p><strong>Data sets are not avaiable as Supplementary Tables.</strong></p>
<h2 id="Tomislav-Domazet-Loso-and-Diethard-Tautz-2008"><a href="#Tomislav-Domazet-Loso-and-Diethard-Tautz-2008" class="headerlink" title="Tomislav Domazet-Lošo and Diethard Tautz, 2008"></a><a href="http://mbe.oxfordjournals.org/content/25/12/2699" target="_blank" rel="external">Tomislav Domazet-Lošo and Diethard Tautz, 2008</a></h2><p><strong>Title</strong>: <em>An Ancient Evolutionary Origin of Genes Associated with Human Genetic Diseases</em></p>
<p>Published <code>Phylostratigraphic Map</code>:</p>
<ul>
<li><strong>Organisms</strong>: <em>Homo sapiens</em> (human); ENSEMBL version 45</li>
<li><strong>E-value cutoff</strong>: 1E-3 (blastp; protein sequences) and 1E-15 (tblastn; ESTs) </li>
<li><strong>Sequence type</strong>: Protein Sequences and ESTs</li>
<li><strong>Reference data bases</strong>: NCBI nr (protein); trace and EST archives (EST)</li>
<li><strong>Splice variants</strong>: always using the longest splice variant</li>
</ul>
<p>Download Map using R:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># download the Phylostratigraphic Map of Homo sapiens</span></span><br><span class="line"><span class="comment"># from Tomislav Domazet-Lošo and Diethard Tautz, 2008</span></span><br><span class="line">download.file( url      = <span class="string">"http://mbe.oxfordjournals.org/content/suppl/2008/09/25/msn214.DC1/mbe-08-0522-File008_msn214.xls"</span>, </span><br><span class="line">               destfile = <span class="string">"MBE_2008_Homo_Sapiens_PhyloMap.xls"</span> )</span><br></pre></td></tr></table></figure>
<p>Read the <code>*.xls</code> file storing the <code>Phylostratigraphic Map</code> of Homo sapiens and format it for the use with <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a>:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># install the readxl package</span></span><br><span class="line">install.packages(<span class="string">"readxl"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">HomoSapiensPhyloMap.MBE &lt;- read_excel(<span class="string">"MBE_2008_Homo_Sapiens_PhyloMap.xls"</span>, sheet = <span class="number">1</span>, skip = <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format Phylostratigraphic Map for use with myTAI</span></span><br><span class="line">HomoSapiens.PhyloMap &lt;- HomoSapiensPhyloMap.MBE[ , c(<span class="string">"Phylostratum"</span>,<span class="string">"Gene_ID"</span>)]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(HomoSapiens.PhyloMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Phylostratum         Gene_ID</span><br><span class="line">1            1 ENSG00000100053</span><br><span class="line">2            1 ENSG00000100058</span><br><span class="line">3            1 ENSG00000206066</span><br><span class="line">4            1 ENSG00000100068</span><br><span class="line">5            1 ENSG00000100077</span><br><span class="line">6            1 ENSG00000133454</span><br></pre></td></tr></table></figure>
<p>Now you can use the <code>MatchMap()</code> function implemented in <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a> to match the <code>Phylostratigraphic Map</code> of Homo sapiens from <a href="http://mbe.oxfordjournals.org/content/25/12/2699" target="_blank" rel="external">Domazet-Lošo and Tautz, 2008</a> to any gene expression set of your interest (see <a href="https://github.com/HajkD/myTAI/blob/master/vignettes/Introduction.Rmd" target="_blank" rel="external">Introduction to Phylotranscriptomics</a> for details).</p>
<h2 id="Tomislav-Domazet-Loso-and-Diethard-Tautz-2010"><a href="#Tomislav-Domazet-Loso-and-Diethard-Tautz-2010" class="headerlink" title="Tomislav Domazet-Lošo and Diethard Tautz, 2010"></a><a href="http://www.biomedcentral.com/1741-7007/8/66" target="_blank" rel="external">Tomislav Domazet-Lošo and Diethard Tautz, 2010</a></h2><p><strong>Title</strong>: <em>Phylostratigraphic tracking of cancer genes suggests a link to the emergence of multicellularity in metazoa</em></p>
<p>Published <code>Phylostratigraphic Map</code>:</p>
<ul>
<li><strong>Organisms</strong>: <em>Homo sapiens</em> (human); based on 20,259 unique proteins published <a href="http://www.sciencemag.org/content/321/5897/1801" target="_blank" rel="external">here</a></li>
<li><strong>E-value cutoff</strong>: 1E-3 (blastp; protein sequences) and 1E-15 (tblastn; ESTs) </li>
<li><strong>Sequence type</strong>: Protein Sequences and ESTs</li>
<li><strong>Reference data bases</strong>: NCBI nr (protein); trace and EST archives (EST)</li>
<li><strong>Splice variants</strong>: always using the longest splice variant</li>
</ul>
<p>Download Map using R:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># download the Phylostratigraphic Map of Homo sapiens</span></span><br><span class="line"><span class="comment"># from Domazet-Lošo and Tautz, 2010</span></span><br><span class="line">download.file( url      = <span class="string">"http://www.biomedcentral.com/content/supplementary/1741-7007-8-66-s1.xls"</span>, </span><br><span class="line">               destfile = <span class="string">"BMCBiology_2010_Homo_Sapiens_PhyloMap.xls"</span> )</span><br></pre></td></tr></table></figure>
<p>Read the <code>*.xls</code> file storing the <code>Phylostratigraphic Map</code> of Homo sapiens and format it for the use with <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a>:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># install the readxl package</span></span><br><span class="line">install.packages(<span class="string">"readxl"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">HomoSapiensPhyloMap.BMCBiology &lt;- read_excel(<span class="string">"BMCBiology_2010_Homo_Sapiens_PhyloMap.xls"</span>, sheet = <span class="number">1</span>, skip = <span class="number">3</span>, col_names = <span class="literal">FALSE</span>)</span><br><span class="line"></span><br><span class="line">colnames(HomoSapiensPhyloMap.BMCBiology)[<span class="number">1</span>:<span class="number">2</span>] &lt;- c(<span class="string">"Gene_ID"</span>,<span class="string">"Phylostratum"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format Phylostratigraphic Map for use with myTAI</span></span><br><span class="line">HomoSapiens.PhyloMap &lt;- HomoSapiensPhyloMap.BMCBiology[ , c(<span class="string">"Phylostratum"</span>,<span class="string">"Gene_ID"</span>)]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(HomoSapiens.PhyloMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Phylostratum      Gene_ID</span><br><span class="line">1            1       15E1.2</span><br><span class="line">2            1       3&apos;HEXO</span><br><span class="line">3            1 A0PJW7_HUMAN</span><br><span class="line">4            1        A2BP1</span><br><span class="line">5            1          A2M</span><br><span class="line">6            1      A3GALT2</span><br></pre></td></tr></table></figure>
<p>Now you can use the <code>MatchMap()</code> function implemented in <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a> to match the <code>Phylostratigraphic Map</code> of Homo sapiens from <a href="http://www.biomedcentral.com/1741-7007/8/66" target="_blank" rel="external">Domazet-Lošo and Tautz, 2010</a> to any gene expression set of your interest (see <a href="https://github.com/HajkD/myTAI/blob/master/vignettes/Introduction.Rmd" target="_blank" rel="external">Introduction to Phylotranscriptomics</a> for details).</p>
<h2 id="Marcel-Quint-Hajk-Georg-Drost-Alexander-Gabel-Kristian-Karsten-Ullrich-Markus-Boenn-Ivo-Grosse-2012"><a href="#Marcel-Quint-Hajk-Georg-Drost-Alexander-Gabel-Kristian-Karsten-Ullrich-Markus-Boenn-Ivo-Grosse-2012" class="headerlink" title="Marcel Quint, Hajk-Georg Drost, Alexander Gabel, Kristian Karsten Ullrich,    Markus Boenn, Ivo Grosse, 2012"></a><a href="http://www.nature.com/nature/journal/v490/n7418/full/nature11394.html" target="_blank" rel="external">Marcel Quint, Hajk-Georg Drost, Alexander Gabel, Kristian Karsten Ullrich,    Markus Boenn, Ivo Grosse, 2012</a></h2><p><strong>Title</strong>: <em>A transcriptomic hourglass in plant embryogenesis</em></p>
<p>Published <code>Phylostratigraphic Map</code>:</p>
<ul>
<li><strong>Organisms</strong>: <em>Arabidopsis thaliana</em> </li>
<li><strong>E-value cutoff</strong>: 1E-5 (blastp; protein sequences) </li>
<li><strong>Sequence type</strong>: Protein Sequences</li>
<li><strong>Reference data bases</strong>: NCBI nr + additional plant genomes (phytozome)</li>
<li><strong>Splice variants</strong>: always using the longest splice variant</li>
</ul>
<p>Published <code>KaKs Maps</code>:</p>
<ul>
<li><strong>Organisms</strong>: <em>Arabidopsis thaliana</em> versus <em>Arabidopsis lyrata</em>; <em>Arabidopsis thaliana</em> versus <em>Brassica rapa</em>; <em>Arabidopsis thaliana</em> versus <em>Capsella rubella</em>; <em>Arabidopsis thaliana</em> versus <em>Thelungiella halophila</em></li>
<li><strong>E-value cutoff</strong>: 1E-5 (blastp - best hit; protein sequences) </li>
<li><strong>Sequence type</strong>: CDS + Protein Sequences</li>
</ul>
<p>Download <code>Phylostratigraphic Map</code> using R:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># download the Phylostratigraphic Map of Arabidopsis thaliana</span></span><br><span class="line"><span class="comment"># from Quint et al., 2012</span></span><br><span class="line">download.file( url      = <span class="string">"http://www.nature.com/nature/journal/v490/n7418/extref/nature11394-s2.xls"</span>, </span><br><span class="line">               destfile = <span class="string">"Nature_2012_Arabidopsis_thaliana_PhyloMap.xls"</span> )</span><br></pre></td></tr></table></figure>
<p>Read the <code>*.xls</code> file storing the <code>Phylostratigraphic Map</code> of Arabidopsis thaliana and format it for the use with <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a>:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># install the readxl package</span></span><br><span class="line">install.packages(<span class="string">"readxl"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">ArabidopsisThalianaPhyloMap.Nature &lt;- read_excel(<span class="string">"Nature_2012_Arabidopsis_thaliana_PhyloMap.xls"</span>, sheet = <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format Phylostratigraphic Map of Arabidopsis thaliana for use with myTAI</span></span><br><span class="line">ArabidopsisThaliana.PhyloMap &lt;- ArabidopsisThalianaPhyloMap.Nature[ , <span class="number">1</span>:<span class="number">2</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(ArabidopsisThaliana.PhyloMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Phylostratum      Gene</span><br><span class="line">1           13 At5g15420</span><br><span class="line">2           13 At2g07719</span><br><span class="line">3           13 At3g43940</span><br><span class="line">4           13 At5g45095</span><br><span class="line">5           13 At5g60260</span><br><span class="line">6           13 At1g54420</span><br></pre></td></tr></table></figure>
<p>Download <code>KaKs Maps</code> using R:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># download the KaKs Maps of Arabidopsis thaliana</span></span><br><span class="line"><span class="comment"># from Quint et al., 2012</span></span><br><span class="line">download.file( url      = <span class="string">"http://www.nature.com/nature/journal/v490/n7418/extref/nature11394-s3.xls"</span>, </span><br><span class="line">               destfile = <span class="string">"Nature_2012_Arabidopsis_thaliana_KaKsMaps.xls"</span> )</span><br><span class="line">```    </span><br><span class="line"></span><br><span class="line"></span><br><span class="line">### _Arabidopsis thaliana_ versus _Arabidopsis lyrata_</span><br><span class="line"></span><br><span class="line">```r</span><br><span class="line"><span class="comment"># install the readxl package</span></span><br><span class="line">install.packages(<span class="string">"readxl"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">Ath_vs_Aly_KaKsMap.Nature &lt;- read_excel(<span class="string">"Nature_2012_Arabidopsis_thaliana_KaKsMaps.xls"</span>, sheet = <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format KaKs Map of Arabidopsis thaliana versus Arabidopsis lyrata for use with myTAI</span></span><br><span class="line">Ath_vs_Aly.KaKsMap &lt;- Ath_vs_Aly_KaKsMap.Nature[ , <span class="number">1</span>:<span class="number">2</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(Ath_vs_Aly.KaKsMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">     KaKs      Gene</span><br><span class="line">1 0.18560 At5g53110</span><br><span class="line">2 0.58980 At2g02320</span><br><span class="line">3 0.07699 At5g14980</span><br><span class="line">4 0.60110 At2g26050</span><br><span class="line">5 0.08910 At4g17650</span><br><span class="line">6 0.23650 At5g20040</span><br></pre></td></tr></table></figure>
<h3 id="Arabidopsis-thaliana-versus-Brassica-rapa"><a href="#Arabidopsis-thaliana-versus-Brassica-rapa" class="headerlink" title="Arabidopsis thaliana versus Brassica rapa"></a><em>Arabidopsis thaliana</em> versus <em>Brassica rapa</em></h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># install the readxl package</span></span><br><span class="line">install.packages(<span class="string">"readxl"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">Ath_vs_Bra_KaKsMap.Nature &lt;- read_excel(<span class="string">"Nature_2012_Arabidopsis_thaliana_KaKsMaps.xls"</span>, sheet = <span class="number">2</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format KaKs Map of Arabidopsis thaliana versus Brassica rapa for use with myTAI</span></span><br><span class="line">Ath_vs_Bra.KaKsMap &lt;- Ath_vs_Bra_KaKsMap.Nature[ , <span class="number">1</span>:<span class="number">2</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(Ath_vs_Bra.KaKsMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">     KaKs      Gene</span><br><span class="line">1 0.22660 At5g53110</span><br><span class="line">2 0.08425 At5g14980</span><br><span class="line">3 0.32040 At2g26050</span><br><span class="line">4 0.24060 At4g17650</span><br><span class="line">5 0.29970 At5g20040</span><br><span class="line">6 0.43220 At4g19750</span><br></pre></td></tr></table></figure>
<h3 id="Arabidopsis-thaliana-versus-Capsella-rubella"><a href="#Arabidopsis-thaliana-versus-Capsella-rubella" class="headerlink" title="Arabidopsis thaliana versus Capsella rubella"></a><em>Arabidopsis thaliana</em> versus <em>Capsella rubella</em></h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># install the readxl package</span></span><br><span class="line">install.packages(<span class="string">"readxl"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">Ath_vs_Crub_KaKsMap.Nature &lt;- read_excel(<span class="string">"Nature_2012_Arabidopsis_thaliana_KaKsMaps.xls"</span>, sheet = <span class="number">3</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format KaKs Map of Arabidopsis thaliana versus Capsella rubella for use with myTAI</span></span><br><span class="line">Ath_vs_Crub.KaKsMap &lt;- Ath_vs_Crub_KaKsMap.Nature[ , <span class="number">1</span>:<span class="number">2</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(Ath_vs_Crub.KaKsMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">     KaKs      Gene</span><br><span class="line">1 0.41030 At1g01010</span><br><span class="line">2 0.50510 At1g01020</span><br><span class="line">3 0.13300 At1g01030</span><br><span class="line">4 0.10300 At1g01040</span><br><span class="line">5 0.04453 At1g01050</span><br><span class="line">6 0.27210 At1g01060</span><br></pre></td></tr></table></figure>
<h3 id="Arabidopsis-thaliana-versus-Thelungiella-halophila"><a href="#Arabidopsis-thaliana-versus-Thelungiella-halophila" class="headerlink" title="Arabidopsis thaliana versus Thelungiella halophila"></a><em>Arabidopsis thaliana</em> versus <em>Thelungiella halophila</em></h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># install the readxl package</span></span><br><span class="line">install.packages(<span class="string">"readxl"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">Ath_vs_Thal_KaKsMap.Nature &lt;- read_excel(<span class="string">"Nature_2012_Arabidopsis_thaliana_KaKsMaps.xls"</span>, sheet = <span class="number">4</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format KaKs Map of Arabidopsis thaliana versus Thelungiella halophila for use with myTAI</span></span><br><span class="line">Ath_vs_Thal.KaKsMap &lt;- Ath_vs_Thal_KaKsMap.Nature[ , <span class="number">1</span>:<span class="number">2</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(Ath_vs_Thal.KaKsMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">     KaKs      Gene</span><br><span class="line">1 0.54110 At1g01010</span><br><span class="line">2 0.37860 At1g01020</span><br><span class="line">3 0.10400 At1g01030</span><br><span class="line">4 0.11070 At1g01040</span><br><span class="line">5 0.01867 At1g01050</span><br><span class="line">6 0.36660 At1g01060</span><br></pre></td></tr></table></figure>
<p>Now you can use the <code>MatchMap()</code> function implemented in <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a> to match the <code>Phylostratigraphic Map</code> and <code>KaKs Maps</code> of Arabidopsis thaliana from <a href="http://www.nature.com/nature/journal/v490/n7418/full/nature11394.html" target="_blank" rel="external">Quint et al., 2012</a> to any gene expression set of your interest (see <a href="https://github.com/HajkD/myTAI/blob/master/vignettes/Introduction.Rmd" target="_blank" rel="external">Introduction to Phylotranscriptomics</a> for details).</p>
<h2 id="Rafik-Neme-and-Diethard-Tautz-2013"><a href="#Rafik-Neme-and-Diethard-Tautz-2013" class="headerlink" title="Rafik Neme and Diethard Tautz, 2013"></a><a href="http://www.biomedcentral.com/1471-2164/14/117" target="_blank" rel="external">Rafik Neme and Diethard Tautz, 2013</a></h2><p><strong>Title</strong>: <em>Phylogenetic patterns of emergence of new genes support a model of frequent de novo evolution</em></p>
<p>Published <code>Phylostratigraphic Map</code>:</p>
<ul>
<li><strong>Organisms</strong>: <em>Mus musculus</em> (Ensembl v. 66), <em>Homo sapiens</em> (Ensembl v. 68), <em>Danio rerio</em> (Ensembl v. 68), <em>Gasterosteus aculeatus</em> (Ensembl v. 68)</li>
<li><strong>E-value cutoff</strong>: 1E-3 (blastp; protein sequences) and 1E-15 (tblastn; ESTs) </li>
<li><strong>Sequence type</strong>: Protein Sequences and ESTs</li>
<li><strong>Reference data bases</strong>: NCBI nr (protein); trace and EST archives (EST)</li>
<li><strong>Splice variants</strong>: always using the longest splice variant</li>
</ul>
<p>Download Map using R:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># download the Phylostratigraphic Maps</span></span><br><span class="line"><span class="comment"># from Neme and Tautz, 2013</span></span><br><span class="line">download.file( url      = <span class="string">"http://www.biomedcentral.com/content/supplementary/1471-2164-14-117-s1.xlsx"</span>, </span><br><span class="line">               destfile = <span class="string">"BMCGenomics_2013_species_PhyloMaps.xlsx"</span> )</span><br></pre></td></tr></table></figure>
<p>Read the <code>*.xlsx</code> file storing the <code>Phylostratigraphic Maps</code> of <em>Mus musculus</em>, <em>Homo sapiens</em>, <em>Danio rerio</em>, and <em>Gasterosteus aculeatus</em> and format it for the use with <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a>:</p>
<h3 id="Mus-musculus"><a href="#Mus-musculus" class="headerlink" title="Mus musculus"></a><em>Mus musculus</em></h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># install the readxl package</span></span><br><span class="line">install.packages(<span class="string">"readxl"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file: Mus musculus</span></span><br><span class="line">MusMusculusPhyloMap.BMCGenomics &lt;- read_excel(<span class="string">"BMCGenomics_2013_species_PhyloMaps.xlsx"</span>, sheet = <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format Phylostratigraphic Map of Mus musculus for use with myTAI</span></span><br><span class="line">MusMusculus.PhyloMap &lt;- MusMusculusPhyloMap.BMCGenomics[ , c(<span class="number">2</span>,<span class="number">1</span>)]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(MusMusculus.PhyloMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Oldest Phylostratum    Ensembl Gene ID</span><br><span class="line">1                   1 ENSMUSG00000074155</span><br><span class="line">2                   1 ENSMUSG00000086875</span><br><span class="line">3                   1 ENSMUSG00000006948</span><br><span class="line">4                   1 ENSMUSG00000079344</span><br><span class="line">5                   1 ENSMUSG00000055193</span><br><span class="line">6                   1 ENSMUSG00000004789</span><br></pre></td></tr></table></figure>
<h3 id="Homo-sapiens"><a href="#Homo-sapiens" class="headerlink" title="Homo sapiens"></a><em>Homo sapiens</em></h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"><span class="comment"># read the excel file: Homo sapiens</span></span><br><span class="line">HomoSapiensPhyloMap.BMCGenomics &lt;- read_excel(<span class="string">"BMCGenomics_2013_species_PhyloMaps.xlsx"</span>, sheet = <span class="number">2</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format Phylostratigraphic Map of Homo sapiens for use with myTAI</span></span><br><span class="line">HomoSapiens.PhyloMap &lt;- HomoSapiensPhyloMap.BMCGenomics[ , c(<span class="number">2</span>,<span class="number">1</span>)]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(HomoSapiens.PhyloMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Oldest Phylostratum Ensembl Gene ID</span><br><span class="line">1                   1 ENSG00000004059</span><br><span class="line">2                   1 ENSG00000004478</span><br><span class="line">3                   1 ENSG00000003137</span><br><span class="line">4                   1 ENSG00000003509</span><br><span class="line">5                   1 ENSG00000001036</span><br><span class="line">6                   1 ENSG00000002587</span><br></pre></td></tr></table></figure>
<h3 id="Danio-rerio"><a href="#Danio-rerio" class="headerlink" title="Danio rerio"></a><em>Danio rerio</em></h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"><span class="comment"># read the excel file: Danio rerio</span></span><br><span class="line">DanioRerioPhyloMap.BMCGenomics &lt;- read_excel(<span class="string">"BMCGenomics_2013_species_PhyloMaps.xlsx"</span>, sheet = <span class="number">3</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format Phylostratigraphic Map of Danio rerio for use with myTAI</span></span><br><span class="line">DanioRerio.PhyloMap &lt;- DanioRerioPhyloMap.BMCGenomics[ , c(<span class="number">2</span>,<span class="number">1</span>)]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(DanioRerio.PhyloMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Oldest Phylostratum    Ensembl Gene ID</span><br><span class="line">1                   1 ENSDARG00000033231</span><br><span class="line">2                   1 ENSDARG00000000102</span><br><span class="line">3                   1 ENSDARG00000000241</span><br><span class="line">4                   1 ENSDARG00000000370</span><br><span class="line">5                   1 ENSDARG00000000380</span><br><span class="line">6                   1 ENSDARG00000000472</span><br></pre></td></tr></table></figure>
<h3 id="Gasterosteus-aculeatus"><a href="#Gasterosteus-aculeatus" class="headerlink" title="Gasterosteus aculeatus"></a><em>Gasterosteus aculeatus</em></h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"><span class="comment"># read the excel file: Gasterosteus aculeatus</span></span><br><span class="line">GasterosteusAculeatusPhyloMap.BMCGenomics &lt;- read_excel(<span class="string">"BMCGenomics_2013_species_PhyloMaps.xlsx"</span>, sheet = <span class="number">4</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format Phylostratigraphic Map of Gasterosteus aculeatus for use with myTAI</span></span><br><span class="line">GasterosteusAculeatus.PhyloMap &lt;- GasterosteusAculeatusPhyloMap.BMCGenomics[ , c(<span class="number">2</span>,<span class="number">1</span>)]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(GasterosteusAculeatus.PhyloMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Oldest Phylostratum    Ensembl Gene ID</span><br><span class="line">1                   1 ENSGACG00000000009</span><br><span class="line">2                   1 ENSGACG00000000017</span><br><span class="line">3                   1 ENSGACG00000000018</span><br><span class="line">4                   1 ENSGACG00000000019</span><br><span class="line">5                   1 ENSGACG00000000022</span><br><span class="line">6                   1 ENSGACG00000000027</span><br></pre></td></tr></table></figure>
<p>Now you can use the <code>MatchMap()</code> function implemented in <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a> to match the <code>Phylostratigraphic Maps</code> of the aforementioned species from <a href="http://www.biomedcentral.com/1471-2164/14/117" target="_blank" rel="external">Neme and Tautz, 2013</a> to any gene expression set of your interest (see <a href="https://github.com/HajkD/myTAI/blob/master/vignettes/Introduction.Rmd" target="_blank" rel="external">Introduction to Phylotranscriptomics</a> for details).</p>
<h2 id="Martin-Sebastijan-Sestak-and-Tomislav-Domazet-Loso-2015"><a href="#Martin-Sebastijan-Sestak-and-Tomislav-Domazet-Loso-2015" class="headerlink" title="Martin Sebastijan Šestak and Tomislav Domazet-Lošo, 2015"></a><a href="http://mbe.oxfordjournals.org/content/32/2/299.short" target="_blank" rel="external">Martin Sebastijan Šestak and Tomislav Domazet-Lošo, 2015</a></h2><p><strong>Title</strong>: <em>Phylostratigraphic Profiles in Zebrafish Uncover Chordate Origins of the Vertebrate Brain</em></p>
<p>Published <code>Phylostratigraphic Map</code>:</p>
<ul>
<li><strong>Organisms</strong>: <em>Danio rerio</em> (zebrafish) and <em>Drosophila melanogaster</em></li>
<li><strong>E-value cutoff</strong>: 1E-3 (blastp; protein sequences) and 1E-15 (tblastn; ESTs) </li>
<li><strong>Sequence type</strong>: Protein Sequences and ESTs</li>
<li><strong>Reference data bases</strong>: NCBI nr (protein); trace and EST archives (EST)</li>
<li><strong>Splice variants</strong>: always using the longest splice variant</li>
</ul>
<p>Download Map using R:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># download the Phylostratigraphic Map of Danio rerio</span></span><br><span class="line"><span class="comment"># from Šestak and Domazet-Lošo, 2015</span></span><br><span class="line">download.file( url      = <span class="string">"http://mbe.oxfordjournals.org/content/suppl/2014/11/17/msu319.DC1/TableS3-2.xlsx"</span>, </span><br><span class="line">               destfile = <span class="string">"MBE_2015a_Drerio_PhyloMap.xlsx"</span> )</span><br><span class="line">               </span><br><span class="line"></span><br><span class="line"><span class="comment"># download the Phylostratigraphic Map of Drosophila melanogaster</span></span><br><span class="line"><span class="comment"># from Šestak and Domazet-Lošo, 2015</span></span><br><span class="line">download.file( url      = <span class="string">"http://mbe.oxfordjournals.org/content/suppl/2014/11/17/msu319.DC1/TableS6.xlsx"</span>, </span><br><span class="line">               destfile = <span class="string">"MBE_2015a_Dmelanogaster_PhyloMap.xlsx"</span> )</span><br></pre></td></tr></table></figure>
<p>Read the <code>*.xlsx</code> file storing the <code>Phylostratigraphic Map</code> of <em>D. rerio</em> and <em>D. melanogaster</em> and format it for the use with <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a>:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># install the readxl package</span></span><br><span class="line">install.packages(<span class="string">"readxl"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">DrerioPhyloMap.MBEa &lt;- read_excel(<span class="string">"MBE_2015a_Drerio_PhyloMap.xlsx"</span>, sheet = <span class="number">1</span>, skip = <span class="number">4</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format Phylostratigraphic Map for use with myTAI</span></span><br><span class="line">Drerio.PhyloMap &lt;- DrerioPhyloMap.MBEa[ , <span class="number">1</span>:<span class="number">2</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(Drerio.PhyloMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Phylostrata            ZFIN_ID</span><br><span class="line">1           1 ZDB-GENE-000208-13</span><br><span class="line">2           1 ZDB-GENE-000208-17</span><br><span class="line">3           1 ZDB-GENE-000208-18</span><br><span class="line">4           1 ZDB-GENE-000208-23</span><br><span class="line">5           1  ZDB-GENE-000209-3</span><br><span class="line">6           1  ZDB-GENE-000209-4</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">DmelanogasterPhyloMap.MBEa &lt;- read_excel(<span class="string">"MBE_2015a_Dmelanogaster_PhyloMap.xlsx"</span>, sheet = <span class="number">1</span>, skip = <span class="number">4</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># format Phylostratigraphic Map for use with myTAI</span></span><br><span class="line">Dmelanogaster.PhyloMap &lt;- DmelanogasterPhyloMap.MBEa[ , <span class="number">1</span>:<span class="number">2</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(Dmelanogaster.PhyloMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Phylostrata FlyBase_Gene_ID</span><br><span class="line">1           1     FBgn0000017</span><br><span class="line">2           1     FBgn0000024</span><br><span class="line">3           1     FBgn0000032</span><br><span class="line">4           1     FBgn0000036</span><br><span class="line">5           1     FBgn0000038</span><br><span class="line">6           1     FBgn0000039</span><br></pre></td></tr></table></figure>
<p>Now you can use the <code>MatchMap()</code> function implemented in <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a> to match the <code>Phylostratigraphic Maps</code> of the aforementioned species from <a href="http://mbe.oxfordjournals.org/content/32/2/299.short" target="_blank" rel="external">Šestak and Domazet-Lošo, 2015</a> to any gene expression set of your interest (see <a href="https://github.com/HajkD/myTAI/blob/master/vignettes/Introduction.Rmd" target="_blank" rel="external">Introduction to Phylotranscriptomics</a> for details).</p>
<h2 id="Hajk-Georg-Drost-Alexander-Gabel-Ivo-Grosse-Marcel-Quint-2015"><a href="#Hajk-Georg-Drost-Alexander-Gabel-Ivo-Grosse-Marcel-Quint-2015" class="headerlink" title="Hajk-Georg Drost, Alexander Gabel, Ivo Grosse, Marcel Quint, 2015"></a><a href="http://mbe.oxfordjournals.org/content/32/5/1221.short?rss=1" target="_blank" rel="external">Hajk-Georg Drost, Alexander Gabel, Ivo Grosse, Marcel Quint, 2015</a></h2><p><strong>Title</strong>: <em>Evidence for Active Maintenance of Phylotranscriptomic Hourglass Patterns in Animal and Plant Embryogenesis</em></p>
<p>Published <code>Phylostratigraphic Map</code>:</p>
<ul>
<li><strong>Organisms</strong>: <em>Danio rerio</em> (zebrafish), <em>Drosophila melanogaster</em> (fly), and <em>Arabidopsis thaliana</em></li>
<li><strong>E-value cutoff</strong>: 1E-5 (blastp; protein sequences)</li>
<li><strong>Sequence type</strong>: Protein Sequences</li>
<li><strong>Reference data bases</strong>: NCBI nr (protein) + custom selection of genomes (phytozome, flybase)</li>
<li><strong>Splice variants</strong>: always using the longest splice variant</li>
</ul>
<p>Published <code>Divergence Maps</code>:</p>
<ul>
<li><strong>Organisms</strong>: <em>Arabidopsis thaliana</em> versus <em>Arabidopsis lyrata</em>; <em>Arabidopsis thaliana</em> versus <em>Brassica rapa</em>; <em>Arabidopsis thaliana</em> versus <em>Capsella rubella</em>; <em>Arabidopsis thaliana</em> versus <em>Thelungiella halophila</em>; <em>Danio rerio</em> versus <em>A. mexicanus</em> ; <em>Danio rerio</em> versus <em>F. rubripes</em>; <em>Danio rerio</em> versus <em>X. maculatus</em> ; <em>Danio rerio</em> versus <em>G. morhua</em> ; <em>Drosophila melanogaster</em> versus <em>D. simulans</em> ; <em>Drosophila melanogaster</em> versus <em>D. yakuba</em> ; <em>Drosophila melanogaster</em> versus <em>D. persimilis</em> ; <em>Drosophila melanogaster</em> versus <em>D. virilis</em> ;</li>
<li><strong>E-value cutoff</strong>: 1E-5 (blastp - best reciprocal hit; protein sequences) </li>
<li><strong>Sequence type</strong>: CDS + Protein Sequences</li>
</ul>
<p>Download Maps using R:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># download the Phylostratigraphic Maps</span></span><br><span class="line"><span class="comment"># from Drost et al., 2015</span></span><br><span class="line">download.file( url      = <span class="string">"http://files.figshare.com/1798295/Supplementary_table_S2.xls"</span>, </span><br><span class="line">               destfile = <span class="string">"MBE_2015b_PhyloMaps.xls"</span> )</span><br><span class="line">               </span><br><span class="line">               </span><br><span class="line"><span class="comment"># download the Divergence Maps</span></span><br><span class="line">download.file( url      = <span class="string">"http://files.figshare.com/1798297/Supplementary_table_S4.xls"</span>, </span><br><span class="line">               destfile = <span class="string">"MBE_2015b_DivergenceMaps.xls"</span> )</span><br></pre></td></tr></table></figure>
<p>Read the <code>*.xls</code> file storing the <code>Phylostratigraphic Maps</code> and <code>Divergence Maps</code> and format it for the use with <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a>:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># install the readxl package</span></span><br><span class="line">install.packages(<span class="string">"readxl"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">DrerioPhyloMap.MBEb &lt;- read_excel(<span class="string">"MBE_2015b_PhyloMaps.xls"</span>, sheet = <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(DrerioPhyloMap.MBEb)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Phylostratum             GeneID</span><br><span class="line">1            1 ENSDARG00000000002</span><br><span class="line">2            1 ENSDARG00000000019</span><br><span class="line">3            1 ENSDARG00000000102</span><br><span class="line">4            1 ENSDARG00000000241</span><br><span class="line">5            1 ENSDARG00000000324</span><br><span class="line">6            1 ENSDARG00000000369</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">DmelanogasterPhyloMap.MBEb &lt;- read_excel(<span class="string">"MBE_2015b_PhyloMaps.xls"</span>, sheet = <span class="number">2</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(DmelanogasterPhyloMap.MBEb)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Phylostratum      GeneID</span><br><span class="line">1            1 fbpp0070006</span><br><span class="line">2            1 fbpp0070025</span><br><span class="line">3            1 fbpp0070051</span><br><span class="line">4            1 fbpp0070054</span><br><span class="line">5            1 fbpp0070061</span><br><span class="line">6            1 fbpp0070064</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># read the excel file</span></span><br><span class="line">Athaliana.MBEb &lt;- read_excel(<span class="string">"MBE_2015b_PhyloMaps.xls"</span>, sheet = <span class="number">3</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(Athaliana.MBEb)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Phylostratum      GeneID</span><br><span class="line">1            1 at1g01040.2</span><br><span class="line">2            1 at1g01050.1</span><br><span class="line">3            1 at1g01070.1</span><br><span class="line">4            1 at1g01080.2</span><br><span class="line">5            1 at1g01090.1</span><br><span class="line">6            1 at1g01120.1</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Danio rerio</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># D. rerio vs. A. mexicanus</span></span><br><span class="line">Drerio_vs_Amex_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">9</span>)</span><br><span class="line"><span class="comment"># D. rerio vs. F. rubripes</span></span><br><span class="line">Drerio_vs_Frubripes_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">10</span>)</span><br><span class="line"><span class="comment"># D. rerio vs. X. maculatus</span></span><br><span class="line">Drerio_vs_Xmac_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">11</span>)</span><br><span class="line"><span class="comment"># D. rerio vs. G. morhua</span></span><br><span class="line">Drerio_vs_Gmor_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">12</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Drosophila melanogaster</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># D. melanogaster vs. D. simulans</span></span><br><span class="line">Dmel_Dsim_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">1</span>)</span><br><span class="line"><span class="comment"># D. melanogaster vs. D. yakuba</span></span><br><span class="line">Dmel_Dyak_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">2</span>)</span><br><span class="line"><span class="comment"># D. melanogaster vs. D. persimilis</span></span><br><span class="line">Dmel_Dper_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">3</span>)</span><br><span class="line"><span class="comment"># D. melanogaster vs. D. virilis</span></span><br><span class="line">Dmel_Dvir_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">4</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Arabidopsis thaliana</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># A. thaliana vs A. lyrata</span></span><br><span class="line">Ath_Aly_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">5</span>)</span><br><span class="line"><span class="comment"># A thaliana vs. T. halophila</span></span><br><span class="line">Ath_Brapa_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">8</span>)</span><br><span class="line"><span class="comment"># A thaliana vs. C. rubella</span></span><br><span class="line">Ath_Crub_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">7</span>)</span><br><span class="line"><span class="comment"># A thaliana vs. C. papaya </span></span><br><span class="line">Ath_Cpapaya_DivergenceExpressionSet &lt;- read_excel(<span class="string">"MBE_2015b_DivergenceMaps.xls"</span>,sheet = <span class="number">6</span>)</span><br></pre></td></tr></table></figure>
<p>Now you can use the <code>MatchMap()</code> function implemented in <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a> to match the <code>Phylostratigraphic Maps</code> and <code>Divergence Maps</code> of the aforementioned species from <a href="http://mbe.oxfordjournals.org/content/32/5/1221.short?rss=1" target="_blank" rel="external">Drost et al., 2015</a> to any gene expression set of your interest (see <a href="https://github.com/HajkD/myTAI/blob/master/vignettes/Introduction.Rmd" target="_blank" rel="external">Introduction to Phylotranscriptomics</a> for details).</p>
<h2 id="Xuanjin-Cheng-Jerome-Ho-Lam-Hui-Yung-Yung-Lee-Patrick-Tik-Wan-Law-and-Hoi-Shan-Kwan-2015"><a href="#Xuanjin-Cheng-Jerome-Ho-Lam-Hui-Yung-Yung-Lee-Patrick-Tik-Wan-Law-and-Hoi-Shan-Kwan-2015" class="headerlink" title="Xuanjin Cheng, Jerome Ho Lam Hui, Yung Yung Lee, Patrick Tik Wan Law and Hoi Shan Kwan, 2015"></a><a href="http://mbe.oxfordjournals.org/content/early/2015/05/08/molbev.msv047" target="_blank" rel="external">Xuanjin Cheng, Jerome Ho Lam Hui, Yung Yung Lee, Patrick Tik Wan Law and Hoi Shan Kwan, 2015</a></h2><p><strong>Title</strong>: <em>A “Developmental Hourglass” in Fungi</em></p>
<p>Published <code>Phylostratigraphic Map</code>:</p>
<ul>
<li><strong>Organisms</strong>: <em>Coprinopsis cinerea</em> (fungi)</li>
<li><strong>E-value cutoff</strong>: 1E-3 (blastp; protein sequences)</li>
<li><strong>Sequence type</strong>: Protein Sequences</li>
<li><strong>Reference data bases</strong>: NCBI nr (protein) + custom selection of genomes</li>
<li><strong>Splice variants</strong>: always using the longest splice variant</li>
</ul>
<p>Download <code>Phylostratigraphic Map</code> and <code>KaKsMaps</code> in R:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># download the Phylostratigraphic Map of Coprinopsis cinerea</span></span><br><span class="line"><span class="comment"># from Cheng et al., 2015</span></span><br><span class="line">download.file( url      = <span class="string">"http://mbe.oxfordjournals.org/content/suppl/2015/02/27/msv047.DC1/Table_S9.xlsx"</span>, </span><br><span class="line">               destfile = <span class="string">"MBE_2015c_Ccinerea_Maps.xlsx"</span> )</span><br></pre></td></tr></table></figure>
<p>Read the <code>*.xls</code> file storing the <code>Phylostratigraphic Maps</code> and <code>Divergence Maps</code> and format it for the use with <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a>:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Coprinopsis cinerea Phylostratigraphic Map</span></span><br><span class="line"></span><br><span class="line">CcinereaPhyloMap.MBEc &lt;- read_excel(<span class="string">"MBE_2015c_Ccinerea_Maps.xlsx"</span>, sheet = <span class="number">2</span>)</span><br><span class="line"></span><br><span class="line">Ccinerea.PhyloMap &lt;- CcinereaPhyloMap.MBEc[ , c(<span class="number">4</span>,<span class="number">1</span>)]</span><br><span class="line"></span><br><span class="line">colnames(Ccinerea.PhyloMap) &lt;- c(<span class="string">"Phylostratum"</span>, <span class="string">"GeneID"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format</span></span><br><span class="line">head(Ccinerea.PhyloMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">  Phylostratum     GeneID</span><br><span class="line">1           10 CC1G_00004</span><br><span class="line">2            4 CC1G_00007</span><br><span class="line">3            3 CC1G_00011</span><br><span class="line">4            2 CC1G_00012</span><br><span class="line">5            3 CC1G_00013</span><br><span class="line">6            2 CC1G_00014</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># load package readxl</span></span><br><span class="line"><span class="keyword">library</span>(readxl)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Coprinopsis cinerea KaKs Maps</span></span><br><span class="line"></span><br><span class="line">CcinereaKaKsMaps.MBEc &lt;- read_excel(<span class="string">"MBE_2015c_Ccinerea_Maps.xlsx"</span>, sheet = <span class="number">3</span>, skip = <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Coprinopsis cinerea versus Agaricus bisporus var bisporus H97</span></span><br><span class="line">Ccin_vs_Abisp.KaKsMap &lt;- CcinereaKaKsMaps.MBEc[ , c(<span class="number">2</span>,<span class="number">1</span>)]</span><br><span class="line"></span><br><span class="line"><span class="comment"># Coprinopsis cinerea versus Laccaria bicolor</span></span><br><span class="line">Ccin_vs_Lbicol.KaKsMap &lt;- CcinereaKaKsMaps.MBEc[ , c(<span class="number">5</span>,<span class="number">1</span>)]</span><br><span class="line"></span><br><span class="line"><span class="comment"># Coprinopsis cinerea versus Lentinula edodes</span></span><br><span class="line">Ccin_vs_Ledodes.KaKsMap &lt;- CcinereaKaKsMaps.MBEc[ , c(<span class="number">8</span>,<span class="number">1</span>)]</span><br><span class="line"></span><br><span class="line"><span class="comment"># Coprinopsis cinerea versus Schizophyllum commune</span></span><br><span class="line">Ccin_vs_Scommune.KaKsMap &lt;- CcinereaKaKsMaps.MBEc[ , c(<span class="number">11</span>,<span class="number">1</span>)]</span><br><span class="line"></span><br><span class="line"><span class="comment"># have a look at the final format: example -&gt; Ccin_vs_Abisp.KaKsMap</span></span><br><span class="line">head(Ccin_vs_Abisp.KaKsMap)</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">                  dN/dS    Gene_id</span><br><span class="line">1   0.20669999999999999 CC1G_00007</span><br><span class="line">2 7.6300000000000007E-2 CC1G_00011</span><br><span class="line">3   0.15290000000000001 CC1G_00012</span><br><span class="line">4                0.3362 CC1G_00013</span><br><span class="line">5                2.7E-2 CC1G_00014</span><br><span class="line">6 3.5700000000000003E-2 CC1G_00015</span><br></pre></td></tr></table></figure>
<p>Now you can use the <code>MatchMap()</code> function implemented in <a href="https://github.com/HajkD/myTAI" target="_blank" rel="external">myTAI</a> to match the <code>Phylostratigraphic Maps</code> and <code>KaKs Maps</code> of the aforementioned species from <a href="http://mbe.oxfordjournals.org/content/early/2015/05/08/molbev.msv047" target="_blank" rel="external">Cheng et al., 2015</a> to any gene expression set of your interest (see <a href="https://github.com/HajkD/myTAI/blob/master/vignettes/Introduction.Rmd" target="_blank" rel="external">Introduction to Phylotranscriptomics</a> for details).</p>
<blockquote>
<p>Github: HajkD/published_phylomaps </p>
<p><img src="https://raw.githubusercontent.com/xieguigang/xieguigang.github.io-hexo/master/images/qrcode/published_phylomaps.png" alt="A Collection of Already Published Phylostratigraphic Maps and Divergence Maps"></p>
</blockquote>

        
        </div>
        <footer class="article-footer">
            <div class="share-container">



</div>

    <a data-url="http://blog.xieguigang.me/2016/07/18/published_phylomaps/" data-id="ciryy8lg600005wv47uifium1" class="article-share-link"><i class="fa fa-share"></i>Share</a>
<script>
    (function ($) {
        // Prevent duplicate binding
        if (typeof(__SHARE_BUTTON_BINDED__) === 'undefined' || !__SHARE_BUTTON_BINDED__) {
            __SHARE_BUTTON_BINDED__ = true;
        } else {
            return;
        }
        $('body').on('click', function() {
            $('.article-share-box.on').removeClass('on');
        }).on('click', '.article-share-link', function(e) {
            e.stopPropagation();

            var $this = $(this),
                url = $this.attr('data-url'),
                encodedUrl = encodeURIComponent(url),
                id = 'article-share-box-' + $this.attr('data-id'),
                offset = $this.offset(),
                box;

            if ($('#' + id).length) {
                box = $('#' + id);

                if (box.hasClass('on')){
                    box.removeClass('on');
                    return;
                }
            } else {
                var html = [
                    '<div id="' + id + '" class="article-share-box">',
                        '<input class="article-share-input" value="' + url + '">',
                        '<div class="article-share-links">',
                            '<a href="https://twitter.com/intent/tweet?url=' + encodedUrl + '" class="fa fa-twitter article-share-twitter" target="_blank" title="Twitter"></a>',
                            '<a href="https://www.facebook.com/sharer.php?u=' + encodedUrl + '" class="fa fa-facebook article-share-facebook" target="_blank" title="Facebook"></a>',
                            '<a href="http://pinterest.com/pin/create/button/?url=' + encodedUrl + '" class="fa fa-pinterest article-share-pinterest" target="_blank" title="Pinterest"></a>',
                            '<a href="https://plus.google.com/share?url=' + encodedUrl + '" class="fa fa-google article-share-google" target="_blank" title="Google+"></a>',
                        '</div>',
                    '</div>'
                ].join('');

              box = $(html);

              $('body').append(box);
            }

            $('.article-share-box.on').hide();

            box.css({
                top: offset.top + 25,
                left: offset.left
            }).addClass('on');

        }).on('click', '.article-share-box', function (e) {
            e.stopPropagation();
        }).on('click', '.article-share-box-input', function () {
            $(this).select();
        }).on('click', '.article-share-box-link', function (e) {
            e.preventDefault();
            e.stopPropagation();

            window.open(this.href, 'article-share-box-window-' + Date.now(), 'width=500,height=450');
        });
    })(jQuery);
</script>

            
    
        <a href="http://blog.xieguigang.me/2016/07/18/published_phylomaps/#comments" class="article-comment-link disqus-comment-count" data-disqus-url="http://blog.xieguigang.me/2016/07/18/published_phylomaps/">Comments</a>
    

        </footer>
    </div>
    
        
<nav id="article-nav">
    
        <a href="/2016/07/26/take_screenshot_login_screen/" id="article-nav-newer" class="article-nav-link-wrap">
            <strong class="article-nav-caption">Newer</strong>
            <div class="article-nav-title">
                
                    Take screenshot of Login Screen in Ubuntu, Elementary &amp; Pear OS
                
            </div>
        </a>
    
    
        <a href="/2016/07/07/china_guangxi_guilin_yangsuo_by_andyanansu-d43r04m/" id="article-nav-older" class="article-nav-link-wrap">
            <strong class="article-nav-caption">Older</strong>
            <div class="article-nav-title">中國·桂林</div>
        </a>
    
</nav>


    
</article>


    
    <section id="comments">
    
        
    <div id="disqus_thread">
        <noscript>Please enable JavaScript to view the <a href="//disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    </div>

    
    </section>

</section>
            
                <aside id="sidebar">
   
        
    
        
    <div class="widget-wrap">
        <h3 class="widget-title">archives</h3>
        <div class="widget">
            <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2016/08/">August 2016</a><span class="archive-list-count">20</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2016/07/">July 2016</a><span class="archive-list-count">5</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2016/06/">June 2016</a><span class="archive-list-count">22</span></li></ul>
        </div>
    </div>

    
        
    <div class="widget-wrap">
        <h3 class="widget-title">tag cloud</h3>
        <div class="widget tagcloud">
            <a href="/tags/3D/" style="font-size: 10px;">3D</a> <a href="/tags/3d/" style="font-size: 10px;">3d</a> <a href="/tags/3d-graphics/" style="font-size: 10px;">3d graphics</a> <a href="/tags/BI/" style="font-size: 10px;">BI</a> <a href="/tags/CLI/" style="font-size: 10px;">CLI</a> <a href="/tags/Chinese/" style="font-size: 10px;">Chinese</a> <a href="/tags/Chinese-rural-dog/" style="font-size: 10px;">Chinese rural dog</a> <a href="/tags/CodeProject/" style="font-size: 15px;">CodeProject</a> <a href="/tags/Compiler/" style="font-size: 10px;">Compiler</a> <a href="/tags/Desktop/" style="font-size: 10px;">Desktop</a> <a href="/tags/Enigma/" style="font-size: 13.33px;">Enigma</a> <a href="/tags/GCModeller/" style="font-size: 11.67px;">GCModeller</a> <a href="/tags/IE/" style="font-size: 10px;">IE</a> <a href="/tags/Linq/" style="font-size: 10px;">Linq</a> <a href="/tags/MSDN/" style="font-size: 10px;">MSDN</a> <a href="/tags/Manhattan-Plot/" style="font-size: 10px;">Manhattan Plot</a> <a href="/tags/Markdown/" style="font-size: 10px;">Markdown</a> <a href="/tags/OMIM/" style="font-size: 10px;">OMIM</a> <a href="/tags/PDB/" style="font-size: 10px;">PDB</a> <a href="/tags/Phylostratigraphic/" style="font-size: 10px;">Phylostratigraphic</a> <a href="/tags/R/" style="font-size: 13.33px;">R</a> <a href="/tags/R-squared/" style="font-size: 10px;">R-squared</a> <a href="/tags/SDK-document/" style="font-size: 10px;">SDK document</a> <a href="/tags/SNP/" style="font-size: 15px;">SNP</a> <a href="/tags/SVG/" style="font-size: 10px;">SVG</a> <a href="/tags/Syntax/" style="font-size: 10px;">Syntax</a> <a href="/tags/UI/" style="font-size: 10px;">UI</a> <a href="/tags/VisualStudio/" style="font-size: 10px;">VisualStudio</a> <a href="/tags/WPF/" style="font-size: 10px;">WPF</a> <a href="/tags/XAML/" style="font-size: 10px;">XAML</a> <a href="/tags/YAML/" style="font-size: 10px;">YAML</a> <a href="/tags/algorithm/" style="font-size: 10px;">algorithm</a> <a href="/tags/architecture/" style="font-size: 11.67px;">architecture</a> <a href="/tags/art-work/" style="font-size: 10px;">art work</a> <a href="/tags/async/" style="font-size: 10px;">async</a> <a href="/tags/bash/" style="font-size: 10px;">bash</a> <a href="/tags/check-point/" style="font-size: 10px;">check-point</a> <a href="/tags/chrome/" style="font-size: 10px;">chrome</a> <a href="/tags/circos/" style="font-size: 11.67px;">circos</a> <a href="/tags/d3js/" style="font-size: 13.33px;">d3js</a> <a href="/tags/data-analysis/" style="font-size: 10px;">data analysis</a> <a href="/tags/data-visualization/" style="font-size: 10px;">data visualization</a> <a href="/tags/desktop/" style="font-size: 10px;">desktop</a> <a href="/tags/document/" style="font-size: 10px;">document</a> <a href="/tags/elementary-os/" style="font-size: 10px;">elementary os</a> <a href="/tags/firefox/" style="font-size: 10px;">firefox</a> <a href="/tags/force-directed/" style="font-size: 10px;">force directed</a> <a href="/tags/gdi/" style="font-size: 11.67px;">gdi+</a> <a href="/tags/genetics/" style="font-size: 13.33px;">genetics</a> <a href="/tags/github/" style="font-size: 13.33px;">github</a> <a href="/tags/heatmap/" style="font-size: 10px;">heatmap</a> <a href="/tags/interface/" style="font-size: 10px;">interface</a> <a href="/tags/javascript/" style="font-size: 16.67px;">javascript</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/language/" style="font-size: 10px;">language</a> <a href="/tags/languages/" style="font-size: 10px;">languages</a> <a href="/tags/libzplay/" style="font-size: 10px;">libzplay</a> <a href="/tags/linux/" style="font-size: 10px;">linux</a> <a href="/tags/math/" style="font-size: 10px;">math</a> <a href="/tags/music/" style="font-size: 16.67px;">music</a> <a href="/tags/my-works/" style="font-size: 10px;">my works</a> <a href="/tags/mysql/" style="font-size: 10px;">mysql</a> <a href="/tags/nodejs/" style="font-size: 10px;">nodejs</a> <a href="/tags/note/" style="font-size: 18.33px;">note</a> <a href="/tags/open-source/" style="font-size: 10px;">open-source</a> <a href="/tags/perl/" style="font-size: 10px;">perl</a> <a href="/tags/pipeline/" style="font-size: 10px;">pipeline</a> <a href="/tags/re0/" style="font-size: 10px;">re0</a> <a href="/tags/rem/" style="font-size: 10px;">rem</a> <a href="/tags/repost/" style="font-size: 16.67px;">repost</a> <a href="/tags/statics/" style="font-size: 10px;">statics</a> <a href="/tags/svg/" style="font-size: 10px;">svg</a> <a href="/tags/threading/" style="font-size: 10px;">threading</a> <a href="/tags/threadpool/" style="font-size: 10px;">threadpool</a> <a href="/tags/tricks/" style="font-size: 10px;">tricks</a> <a href="/tags/ubuntu/" style="font-size: 10px;">ubuntu</a> <a href="/tags/user-agent/" style="font-size: 10px;">user-agent</a> <a href="/tags/vb-net/" style="font-size: 20px;">vb.net</a> <a href="/tags/web/" style="font-size: 10px;">web</a> <a href="/tags/web-api/" style="font-size: 10px;">web api</a> <a href="/tags/web-server/" style="font-size: 10px;">web server</a> <a href="/tags/win10/" style="font-size: 10px;">win10</a> <a href="/tags/works/" style="font-size: 10px;">works</a> <a href="/tags/中華田園犬/" style="font-size: 10px;">中華田園犬</a> <a href="/tags/前端/" style="font-size: 10px;">前端</a> <a href="/tags/前端设计/" style="font-size: 13.33px;">前端设计</a> <a href="/tags/数据可视化/" style="font-size: 10px;">数据可视化</a> <a href="/tags/数据视图/" style="font-size: 10px;">数据视图</a> <a href="/tags/架构/" style="font-size: 11.67px;">架构</a> <a href="/tags/架构设计/" style="font-size: 10px;">架构设计</a> <a href="/tags/线程池/" style="font-size: 10px;">线程池</a> <a href="/tags/转载/" style="font-size: 10px;">转载</a> <a href="/tags/转载搬运/" style="font-size: 16.67px;">转载搬运</a> <a href="/tags/连锁不平衡/" style="font-size: 11.67px;">连锁不平衡</a> <a href="/tags/遗传学/" style="font-size: 11.67px;">遗传学</a> <a href="/tags/鹿乃/" style="font-size: 10px;">鹿乃</a>
        </div>
    </div>

    
        
    <div class="widget-wrap widget-list">
        <h3 class="widget-title">links</h3>
        <div class="widget">
            <ul>
                
                    <li>
                        <a href="http://framework-docs.xieguigang.me/">VisualBasic Framework</a>
                    </li>
                
                    <li>
                        <a href="http://hexo.io">Hexo</a>
                    </li>
                
                    <li>
                        <a href="http://gcmodeller.org">http://gcmodeller.org</a>
                    </li>
                
                    <li>
                        <a href="https://github.com/genetics-potato">https://github.com/genetics-potato</a>
                    </li>
                
            </ul>
        </div>
    </div>


    
    <div id="toTop" class="fa fa-angle-up"></div>
</aside>
            
        </div>
        <footer id="footer">
    <div class="outer">
        <div id="footer-info" class="inner">
            &copy; 2016 しゃけい よしつな<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>. Theme by <a href="http://github.com/ppoffice">PPOffice</a>
        </div>
    </div>
</footer>
        
    
    <script>
    var disqus_config = function () {
        
            this.page.url = 'http://blog.xieguigang.me/2016/07/18/published_phylomaps/';
        
        this.page.identifier = 'published_phylomaps';
    };
    (function() { 
        var d = document, s = d.createElement('script');  
        s.src = '//' + 'xieguigang' + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>



    
        <script src="/vendor/fancybox/jquery.fancybox.pack.js"></script>
    


<!-- Custom Scripts -->
<script src="/js/main.js"></script>

    </div>
</body>
</html>