<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  <meta http-equiv="X-UA-Compatible" content="IE=edge" >
  <title>Calculating F-statistics from genotypic data | 拾荒者</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="Genotype


Populations
AA
Aa
aa




Subpopulation 1
125
250
125


Subpopulation 2
50
30
20


Subpopulation 3
100
500
400



N (number of individuals genotyped. The sum of each of the rows in the table">
<meta property="og:type" content="article">
<meta property="og:title" content="Calculating F-statistics from genotypic data">
<meta property="og:url" content="http://yoursite.com/2016/08/03/fst/index.html">
<meta property="og:site_name" content="拾荒者">
<meta property="og:description" content="Genotype


Populations
AA
Aa
aa




Subpopulation 1
125
250
125


Subpopulation 2
50
30
20


Subpopulation 3
100
500
400



N (number of individuals genotyped. The sum of each of the rows in the table">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p1.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p2.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p3.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp1.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp2.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp3.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/Fs.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HI.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HS.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HT.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FIS.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FST.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FIT.png">
<meta property="og:updated_time" content="2016-08-02T23:06:35.500Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Calculating F-statistics from genotypic data">
<meta name="twitter:description" content="Genotype


Populations
AA
Aa
aa




Subpopulation 1
125
250
125


Subpopulation 2
50
30
20


Subpopulation 3
100
500
400



N (number of individuals genotyped. The sum of each of the rows in the table">
<meta name="twitter:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p1.png">
  
    <link rel="alternative" href="/atom.xml" title="拾荒者" type="application/atom+xml">
  
  
    <link rel="icon" href="http://xieguigang.me/favicon.ico">
  
  <link rel="stylesheet" href="/css/style.css">
</head>

<body>
  <div id="container">
    <div class="left-col">
    <div class="overlay"></div>
<div class="intrude-less">
	<header id="header" class="inner">
		<a href="/" class="profilepic">
			
			<img lazy-src="https://avatars1.githubusercontent.com/u/9410171?v=3&amp;s=460" class="js-avatar">
			
		</a>

		<hgroup>
		  <h1 class="header-author"><a href="/">しゃけい よしつな</a></h1>
		</hgroup>

		

		
			<div class="switch-btn">
				<div class="icon">
					<div class="icon-ctn">
						<div class="icon-wrap icon-house" data-idx="0">
							<div class="birdhouse"></div>
							<div class="birdhouse_holes"></div>
						</div>
						<div class="icon-wrap icon-ribbon hide" data-idx="1">
							<div class="ribbon"></div>
						</div>
						
						<div class="icon-wrap icon-link hide" data-idx="2">
							<div class="loopback_l"></div>
							<div class="loopback_r"></div>
						</div>
						
						
						<div class="icon-wrap icon-me hide" data-idx="3">
							<div class="user"></div>
							<div class="shoulder"></div>
						</div>
						
					</div>
					
				</div>
				<div class="tips-box hide">
					<div class="tips-arrow"></div>
					<ul class="tips-inner">
						<li>Menu</li>
						<li>Tags</li>
						
						<li>Links</li>
						
						
						<li>Über</li>
						
					</ul>
				</div>
			</div>
		

		<div class="switch-area">
			<div class="switch-wrap">
				<section class="switch-part switch-part1">
					<nav class="header-menu">
						<ul>
						
							<li><a href="http://xieguigang.me/">HOME</a></li>
				        
							<li><a href="/archives">Archives</a></li>
				        
							<li><a href="/nuget">Works</a></li>
				        
							<li><a href="/about">About</a></li>
				        
						</ul>
					</nav>
					<nav class="header-nav">
						<div class="social">
							
								<a class="github" target="_blank" href="https://github.com/xieguigang" title="github">github</a>
					        
								<a class="weibo" target="_blank" href="http://weibo.com/ineat" title="weibo">weibo</a>
					        
								<a class="mail" target="_blank" href="mailto://I@xieguigang.me" title="mail">mail</a>
					        
								<a class="google" target="_blank" href="/xie.guigang@gmail.com" title="google">google</a>
					        
								<a class="twitter" target="_blank" href="https://twitter.com/xieguigang" title="twitter">twitter</a>
					        
						</div>
					</nav>
				</section>
				
				
				<section class="switch-part switch-part2">
					<div class="widget tagcloud" id="js-tagcloud">
						<a href="/tags/3d-graphics/" style="font-size: 10px;">3d graphics</a> <a href="/tags/CodeProject/" style="font-size: 16px;">CodeProject</a> <a href="/tags/Compiler/" style="font-size: 10px;">Compiler</a> <a href="/tags/Desktop/" style="font-size: 10px;">Desktop</a> <a href="/tags/GCModeller/" style="font-size: 10px;">GCModeller</a> <a href="/tags/Linq/" style="font-size: 10px;">Linq</a> <a href="/tags/Markdown/" style="font-size: 10px;">Markdown</a> <a href="/tags/R/" style="font-size: 10px;">R</a> <a href="/tags/SDK-document/" style="font-size: 10px;">SDK document</a> <a href="/tags/Syntax/" style="font-size: 10px;">Syntax</a> <a href="/tags/UI/" style="font-size: 10px;">UI</a> <a href="/tags/VisualStudio/" style="font-size: 10px;">VisualStudio</a> <a href="/tags/WPF/" style="font-size: 10px;">WPF</a> <a href="/tags/XAML/" style="font-size: 10px;">XAML</a> <a href="/tags/YAML/" style="font-size: 10px;">YAML</a> <a href="/tags/algorithm/" style="font-size: 10px;">algorithm</a> <a href="/tags/architecture/" style="font-size: 12px;">architecture</a> <a href="/tags/bash/" style="font-size: 10px;">bash</a> <a href="/tags/check-point/" style="font-size: 10px;">check-point</a> <a href="/tags/circos/" style="font-size: 12px;">circos</a> <a href="/tags/d3js/" style="font-size: 12px;">d3js</a> <a href="/tags/data-visualization/" style="font-size: 10px;">data visualization</a> <a href="/tags/desktop/" style="font-size: 10px;">desktop</a> <a href="/tags/document/" style="font-size: 10px;">document</a> <a href="/tags/elementary-os/" style="font-size: 10px;">elementary os</a> <a href="/tags/genetics/" style="font-size: 10px;">genetics</a> <a href="/tags/github/" style="font-size: 12px;">github</a> <a href="/tags/language/" style="font-size: 10px;">language</a> <a href="/tags/libzplay/" style="font-size: 10px;">libzplay</a> <a href="/tags/music/" style="font-size: 14px;">music</a> <a href="/tags/my-works/" style="font-size: 10px;">my works</a> <a href="/tags/nodejs/" style="font-size: 10px;">nodejs</a> <a href="/tags/note/" style="font-size: 18px;">note</a> <a href="/tags/perl/" style="font-size: 10px;">perl</a> <a href="/tags/re0/" style="font-size: 10px;">re0</a> <a href="/tags/rem/" style="font-size: 10px;">rem</a> <a href="/tags/repost/" style="font-size: 12px;">repost</a> <a href="/tags/statics/" style="font-size: 10px;">statics</a> <a href="/tags/svg/" style="font-size: 10px;">svg</a> <a href="/tags/tricks/" style="font-size: 10px;">tricks</a> <a href="/tags/ubuntu/" style="font-size: 10px;">ubuntu</a> <a href="/tags/user-agent/" style="font-size: 10px;">user-agent</a> <a href="/tags/vb-net/" style="font-size: 20px;">vb.net</a> <a href="/tags/web/" style="font-size: 10px;">web</a> <a href="/tags/win10/" style="font-size: 10px;">win10</a> <a href="/tags/works/" style="font-size: 10px;">works</a> <a href="/tags/前端/" style="font-size: 10px;">前端</a> <a href="/tags/前端设计/" style="font-size: 14px;">前端设计</a> <a href="/tags/搬运转载/" style="font-size: 10px;">搬运转载</a> <a href="/tags/架构/" style="font-size: 10px;">架构</a> <a href="/tags/架构设计/" style="font-size: 10px;">架构设计</a> <a href="/tags/转载/" style="font-size: 10px;">转载</a> <a href="/tags/转载搬运/" style="font-size: 16px;">转载搬运</a>
					</div>
				</section>
				
				
				
				<section class="switch-part switch-part3">
					<div id="js-friends">
					
			          <a target="_blank" class="main-nav-link switch-friends-link" href="http://codeproject.com/">Code Project</a>
			        
			          <a target="_blank" class="main-nav-link switch-friends-link" href="http://gcmodeller.org/">GCModeller</a>
			        
			        </div>
				</section>
				

				
				
				<section class="switch-part switch-part4">
				
					<div id="js-aboutme">大神在此！</div>
				</section>
				
			</div>
		</div>
	</header>				
</div>

    </div>
    <div class="mid-col">
      <nav id="mobile-nav">
  	<div class="overlay">
  		<div class="slider-trigger"></div>
  		<h1 class="header-author js-mobile-header hide">しゃけい よしつな</h1>
  	</div>
	<div class="intrude-less">
		<header id="header" class="inner">
			<div class="profilepic">
			
				<img lazy-src="https://avatars1.githubusercontent.com/u/9410171?v=3&amp;s=460" class="js-avatar">
			
			</div>
			<hgroup>
			  <h1 class="header-author">しゃけい よしつな</h1>
			</hgroup>
			
			<nav class="header-menu">
				<ul>
				
					<li><a href="http://xieguigang.me/">HOME</a></li>
		        
					<li><a href="/archives">Archives</a></li>
		        
					<li><a href="/nuget">Works</a></li>
		        
					<li><a href="/about">About</a></li>
		        
		        <div class="clearfix"></div>
				</ul>
			</nav>
			<nav class="header-nav">
				<div class="social">
					
						<a class="github" target="_blank" href="https://github.com/xieguigang" title="github">github</a>
			        
						<a class="weibo" target="_blank" href="http://weibo.com/ineat" title="weibo">weibo</a>
			        
						<a class="mail" target="_blank" href="mailto://I@xieguigang.me" title="mail">mail</a>
			        
						<a class="google" target="_blank" href="/xie.guigang@gmail.com" title="google">google</a>
			        
						<a class="twitter" target="_blank" href="https://twitter.com/xieguigang" title="twitter">twitter</a>
			        
				</div>
			</nav>
		</header>				
	</div>
</nav>

      <div class="body-wrap"><article id="post-fst" class="article article-type-post" itemscope itemprop="blogPost">
  
    <div class="article-meta">
      <a href="/2016/08/03/fst/" class="article-date">
  	<time datetime="2016-08-02T16:00:00.000Z" itemprop="datePublished">2016-08-03</time>
</a>
    </div>
  
  <div class="article-inner">
    
      <input type="hidden" class="isFancy" />
    
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      Calculating F-statistics from genotypic data
    </h1>
  

      </header>
      
      <div class="article-info article-info-post">
        
	<div class="article-tag tagcloud">
		<ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/algorithm/">algorithm</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/genetics/">genetics</a></li></ul>
	</div>

        

        <div class="clearfix"></div>
      </div>
      
    
    <div class="article-entry" itemprop="articleBody">
      
        <h6 id="Genotype"><a href="#Genotype" class="headerlink" title="Genotype"></a>Genotype</h6><table>
<thead>
<tr>
<th>Populations</th>
<th>AA</th>
<th>Aa</th>
<th>aa</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subpopulation 1</td>
<td>125</td>
<td>250</td>
<td>125</td>
</tr>
<tr>
<td>Subpopulation 2</td>
<td>50</td>
<td>30</td>
<td>20</td>
</tr>
<tr>
<td>Subpopulation 3</td>
<td>100</td>
<td>500</td>
<td>400</td>
</tr>
</tbody>
</table>
<p><em>N</em> (number of individuals genotyped. The sum of each of the rows in the table above):</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Population 1:   500</span><br><span class="line">Population 2:   100</span><br><span class="line">Population 3:   1,000</span><br></pre></td></tr></table></figure>
<p>Remember that the number of alleles is <strong>TWICE</strong> the number of genotypes.</p>
<a id="more"></a>
<p><strong>Step 1.  Calculate the gene (allele) frequencies</strong></p>
<p>Each homozygote will have two alleles, each heterozygote will have one allele.  Note that the denominator will be twice <code>Ni</code> (twice as many alleles as individuals).</p>
<blockquote>
<p><strong>Eqns FST.1</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p1.png" alt=""></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p2.png" alt=""></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p3.png" alt=""></p>
<ul>
<li><code>p1</code> means the frequency of allele <code>A</code> in population 1.</li>
</ul>
</blockquote>
<p><strong>Step 2.  Calculate the expected genotypic counts under Hardy-Weinberg Equilibrium</strong>, and then calculate the <strong>excess or deficiency of homozygotes in each subpopulation</strong>:</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">Pop. 1     Expected AA = 500*0.5^2      = 125     (= observed)</span><br><span class="line">           Expected Aa = 500*2*0.5*0.5  = 250     (= observed)</span><br><span class="line">           Expected aa = 500*0.5^2      = 125     (= observed)</span><br><span class="line"></span><br><span class="line">Pop. 2     Expected AA = 100*0.65^2        = 42.25   (observed has excess of 7.75)</span><br><span class="line">           Expected Aa = 100*2*0.65*0.35   =  45.5   (observed has deficit of 15.5)</span><br><span class="line">           Expected aa = 100*0.352         = 12.25   (observed has excess of 7.75)</span><br></pre></td></tr></table></figure>
<p>Note that sum of two types of homozygote excess = amount of heterozygote deficiency. These quantities have to balance (it’s a mathematical necessity, given that <code>p + q =1</code>.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Pop. 3     Expected AA = 1,000*0.35^2      = 122.5  (observed has deficiency of 22.5)</span><br><span class="line">           Expected Aa = 1,000*2*0.65*0.35 =  455   (observed has excess of 45)</span><br><span class="line">           Expected aa = 1,000*0.352       = 422.5  (observed has deficiency of 22.5)</span><br></pre></td></tr></table></figure>
<p>Summary of homozygote deficiency or excess relative to <code>HWE</code>:</p>
<ul>
<li>Pop. 1.  Observed = Expected: perfect fit</li>
<li>Pop. 2.  Excess of 15.5 homozygotes: some inbreeding</li>
<li>Pop. 3.  Deficiency of 45 homozygotes: outbred or experiencing a Wahlund effect (isolate breaking).</li>
</ul>
<p><strong>Step 3.  Calculate the local <em>observed</em> heterozygosities</strong> of each subpopulation (we will call them Hobs s, where the s  subscript refers to the <code>sth</code> of <code>n</code> populations  – 3 in this example).<br>Here we count genotypes:</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Hobs 1 = 250/500  = 0.5</span><br><span class="line">Hobs 2 =  30/100  = 0.3</span><br><span class="line">Hobs 3 = 500/1000 = 0.5</span><br></pre></td></tr></table></figure>
<p><strong>Step 4.  Calculate the local expected heterozygosity, or gene diversity, of each subpopulation</strong> (modified version of Eqn 35.1):</p>
<blockquote>
<p><strong>Eqns FST.2</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp1.png" alt=""><br><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp2.png" alt=""><br><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp3.png" alt=""></p>
<p>(With two alleles it would actually be easier to use <code>2pq</code> than to use the summation format of Eqn 33.1)</p>
</blockquote>
<p><strong>Notation</strong>: Note that I am using <code>p1</code> and <code>q1</code> here (where the subscripts refer to subpopulations 1 through 3). We would need to use multiple subscripts if we were using the notation of Eqn 35.1 where the alleles are <code>pi</code> (and the i refer to alleles 1 to k).  Indeed, with real multi-locus multipopulation data, we would have a triple summation and three subscripts; one for alleles (<code>i =1 to k</code>), one for the loci (l =1 tom), and one for subpopulations (s = 1 to n).</p>
<p><strong>Step 5.  Calculate the local inbreeding coefficient of each subpopulation</strong> (same as Eqn 35.4, except that we are subscripting for the subpopulations):</p>
<blockquote>
<p><strong>Eqn FST.3</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/Fs.png" alt=""></p>
<p>where <em>s</em> (<code>s = 1 to 3</code>) refers to the subpopulation</p>
</blockquote>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">F1 = (0.5 - 0.5) / 0.5     =  0</span><br><span class="line">F2 = (0.455 - 0.3) / 0.455 =  0.341 [positive F means fewer heterozygotes than expected indicates inbreeding]</span><br><span class="line">F3 = (0.455 — 0.5) / 0.455 = -0.099 [negative F means more heterozygotes than expected means excess outbreeding]</span><br></pre></td></tr></table></figure>
<p><strong>Step 6. Calculate <code>p</code> (p-bar, the frequency of allele A) over the total population.</strong><br>[Note that if we had more alleles we could put this and Step 7 all together as a single “global gene frequencies” step, or have one for each allele frequency].</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">(2*125+250+2*50+30+2*100+500)/(1000+200+2000)=0.4156  &#123;genotype splitting method&#125;</span><br></pre></td></tr></table></figure>
<p>or (yields same answer)<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">(0.5*1000+0.65*200+0.35*2000)/(1000+200+2000)=0.4156  &#123;using Eqn FST.1 values for ps&#125;</span><br></pre></td></tr></table></figure></p>
<p>Note that we weight by <strong>population size</strong></p>
<p><strong>Step 7.  Calculate <code>q</code> (q-bar, the frequency of allele a) over the total population</strong></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">(2*125+250+2*20+30+2*400+500)/(1000+200+2000)=0.5844</span><br></pre></td></tr></table></figure>
<p>Check: <code>p-bar + q-bar = 0.4156 + 0.5844 = 1.0</code> (as required by Eqn 31.1).<br>The check doesn’t guarantee that our result is correct, but if they don’t sum to one, we know we miscalculated.</p>
<p><strong>Step 8.  Calculate the global heterozygosity indices (over <em>I</em>ndividuals, <em>S</em>ubpopulations and <em>T</em>otal population)</strong><br>Note that the first two calculations employ a weighted average of the values in the whole set of subpopulations.</p>
<p><code>HI</code> based on <strong>observed</strong> heterozygosities in <strong>individuals</strong> in subpopulations</p>
<blockquote>
<p><strong>Eqn FST.4</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HI.png" alt=""></p>
</blockquote>
<p><code>HS</code> based on <strong>expected</strong> heterozygosities in <strong>subpopulations</strong></p>
<blockquote>
<p><strong>Eqn FST.5</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HS.png" alt=""></p>
</blockquote>
<p><code>HT</code> based on <strong>expected</strong> heterozygosities for overall total population (using global allele frequencies and a modified form of Eqn 35.1):</p>
<blockquote>
<p><strong>Eqn FST.6</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HT.png" alt=""></p>
</blockquote>
<p>or we could calculate it as <code>2*p-bar *q-bar   = 2 * 0.4156 * 0.5844  = 0.4858</code></p>
<p><strong>Step 9.  CALCULATE THE GLOBAL <code>F-STATISTICS</code></strong><br>Compare and contrast the global FISbelow with the <strong>local inbreeding coefficient</strong> <code>Fs</code> of Step 5.<br>Here we are using a weighted average of the individual heterozygosities over all the subpopulations.<br>Both <code>FIS</code> and <code>Fs</code> are, however, based on the <code>observed</code> heterozygosities, whereas <code>FST</code> and <code>FIT</code> are based   on expected heterozygosities.</p>
<blockquote>
<p><strong>Eqn FST.7</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FIS.png" alt=""></p>
<p><strong>Eqn FST.8</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FST.png" alt=""></p>
<p><strong>Eqn FST.9</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FIT.png" alt=""></p>
</blockquote>
<p><strong>Notation note</strong>: the subscripts <code>I</code>, <code>S</code>, and <code>T</code> are not summation subscripts. They simply indicate the level of our analysis.  Likewise, the <code>s</code> on <code>Fs</code> in Step 5 or on the <code>ps</code> in Step 1 (the <code>s</code> was implicit there) just tell us what we are referring to. In contrast, the subscripts for <code>Eqn 35.1</code> and <code>35.2</code> are used in summations and change as we work through the pieces of the calculation.</p>
<p><strong>Step 10.  Finally, draw some conclusions about the genetic structure of the population and its subpopulations</strong>.</p>
<ul>
<li>One of the possible HWE conclusions we could make:<ul>
<li>Pop. 1 is consistent with <code>HWE</code> (results of Step 2)</li>
</ul>
</li>
<li>Two of the possible “local inbreeding” conclusions we could make from Step 5:<ul>
<li>Pop. 2 is inbred (results of Step 5), and</li>
<li>Pop. 3 may have disassortative mating or be experiencing a Wahlund effect (more heterozygotes than expected).</li>
</ul>
</li>
<li>Conclusion concerning overall degree of genetic differentiation (<code>FST</code>)<ul>
<li>Subdivision of populations, possibly due to genetic drift, accounts for approx. 3.4% of the total genetic variation (result of Eqn FST.8 FST calculation in Step 9),</li>
</ul>
</li>
<li>No excess or deficiency of heterozygotes over the total population (<code>FIT</code> is nearly zero).</li>
</ul>
<blockquote>
<p>source: <a href="http://www.uwyo.edu/dbmcd/popecol/index.html" target="_blank" rel="external">Population Ecology, ZOO 4400/5400, Spring 2013</a></p>
</blockquote>

      
    </div>
    
  </div>
  
    
<nav id="article-nav">
  
  
    <a href="/2016/08/02/MCMXC_A_D/" id="article-nav-older" class="article-nav-link-wrap">
      <div class="article-nav-title">MCMXC, A.D.</div>
      <strong class="article-nav-caption">></strong>
    </a>
  
</nav>

  
</article>








<section id="comments">
  <div id="disqus_thread"></div>
    <script type="text/javascript">
    /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
    var disqus_shortname = 'xieguigang'; // required: replace example with your forum shortname

    /* * * DON'T EDIT BELOW THIS LINE * * */
    (function() {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</section>

</div>
      <footer id="footer">
  <div class="outer">
    <div id="footer-info">
    	<div class="footer-left">
    		&copy; 2016 しゃけい よしつな
    	</div>
      	<div class="footer-right">
      		<a href="http://hexo.io/" target="_blank">Hexo</a>  Theme <a href="https://github.com/litten/hexo-theme-yilia" target="_blank">Yilia</a> by Litten
      	</div>
    </div>
  </div>
</footer>
    </div>
    
  <link rel="stylesheet" href="/fancybox/jquery.fancybox.css">


<script>
	var yiliaConfig = {
		fancybox: true,
		mathjax: true,
		animate: true,
		isHome: false,
		isPost: true,
		isArchive: false,
		isTag: false,
		isCategory: false,
		open_in_new: false
	}
</script>
<script src="http://7.url.cn/edu/jslib/comb/require-2.1.6,jquery-1.9.1.min.js"></script>
<script src="/js/main.js"></script>






<script type="text/x-mathjax-config">
MathJax.Hub.Config({
    tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
        processEscapes: true,
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
    }
});

MathJax.Hub.Queue(function() {
    var all = MathJax.Hub.getAllJax(), i;
    for(i=0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';                 
    }       
});
</script>

<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>


  </div>
</body>
</html>