<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    
    <title>Calculating F-statistics from genotypic data | 拾荒者</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
    <meta name="description" content="CLI_tools source code on github: pairwise.fst

Genotype


Populations
AA
Aa
aa




Subpopulation 1
125
250
125


Subpopulation 2
50
30
20


Subpopulation 3
100
500
400



N (number of individuals gen">
<meta property="og:type" content="article">
<meta property="og:title" content="Calculating F-statistics from genotypic data">
<meta property="og:url" content="http://blog.xieguigang.me/2016/08/03/fst/index.html">
<meta property="og:site_name" content="拾荒者">
<meta property="og:description" content="CLI_tools source code on github: pairwise.fst

Genotype


Populations
AA
Aa
aa




Subpopulation 1
125
250
125


Subpopulation 2
50
30
20


Subpopulation 3
100
500
400



N (number of individuals gen">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p1.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p2.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p3.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp1.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp2.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp3.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/Fs.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HI.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HS.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HT.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FIS.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FST.png">
<meta property="og:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FIT.png">
<meta property="og:image" content="https://raw.githubusercontent.com/xieguigang/xieguigang.github.io-hexo/master/images/qrcode/Calculating_F-statistics_from_genotypic_data.png">
<meta property="og:updated_time" content="2016-08-04T20:24:03.134Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Calculating F-statistics from genotypic data">
<meta name="twitter:description" content="CLI_tools source code on github: pairwise.fst

Genotype


Populations
AA
Aa
aa




Subpopulation 1
125
250
125


Subpopulation 2
50
30
20


Subpopulation 3
100
500
400



N (number of individuals gen">
<meta name="twitter:image" content="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p1.png">
    

    
        <link rel="alternate" href="/" title="拾荒者" type="application/atom+xml" />
    

    
        <link rel="icon" href="http://xieguigang.me/favicon.ico" />
    

    <link rel="stylesheet" href="/vendor/font-awesome/css/font-awesome.min.css">
    <link rel="stylesheet" href="/vendor/open-sans/styles.css">
    <link rel="stylesheet" href="/vendor/source-code-pro/styles.css">

    <link rel="stylesheet" href="/css/style.css">

    <script src="/vendor/jquery/2.1.3/jquery.min.js"></script>
    
    
        <link rel="stylesheet" href="/vendor/fancybox/jquery.fancybox.css">
    
    
    
    

</head>

<body>
    <div id="container">
        <header id="header">
    <div id="header-main" class="header-inner">
        <div class="outer">
            <a href="/" id="logo">
                <i class="logo"></i>
                <span class="site-title">拾荒者</span>
            </a>
            <nav id="main-nav">
                
                    <a class="main-nav-link" href="/.">Home</a>
                
                    <a class="main-nav-link" href="/archives">Archives</a>
                
                    <a class="main-nav-link" href="/nuget">nuget</a>
                
                    <a class="main-nav-link" href="/about">About</a>
                
            </nav>
            
                
                <nav id="sub-nav">
                    <div class="profile" id="profile-nav">
                        <a id="profile-anchor" href="javascript:;">
                            <img class="avatar" src="https://raw.githubusercontent.com/xieguigang/xieguigang.github.io-hexo/master/images/my/WeChatImage636059681622385094.jpg" />
                            <i class="fa fa-caret-down"></i>
                        </a>
                    </div>
                </nav>
            
            <div id="search-form-wrap">

    <form class="search-form">
        <input type="text" class="ins-search-input search-form-input" placeholder="Search" />
        <button type="submit" class="search-form-submit"></button>
    </form>
    <div class="ins-search">
    <div class="ins-search-mask"></div>
    <div class="ins-search-container">
        <div class="ins-input-wrapper">
            <input type="text" class="ins-search-input" placeholder="Type something..." />
            <span class="ins-close ins-selectable"><i class="fa fa-times-circle"></i></span>
        </div>
        <div class="ins-section-wrapper">
            <div class="ins-section-container"></div>
        </div>
    </div>
</div>
<script>
(function (window) {
    var INSIGHT_CONFIG = {
        TRANSLATION: {
            POSTS: 'Posts',
            PAGES: 'Pages',
            CATEGORIES: 'Categories',
            TAGS: 'Tags',
            UNTITLED: '(Untitled)',
        },
        ROOT_URL: '/',
        CONTENT_URL: '/content.json',
    };
    window.INSIGHT_CONFIG = INSIGHT_CONFIG;
})(window);
</script>
<script src="/js/insight.js"></script>

</div>
        </div>
    </div>
    <div id="main-nav-mobile" class="header-sub header-inner">
        <table class="menu outer">
            <tr>
                
                    <td><a class="main-nav-link" href="/.">Home</a></td>
                
                    <td><a class="main-nav-link" href="/archives">Archives</a></td>
                
                    <td><a class="main-nav-link" href="/nuget">nuget</a></td>
                
                    <td><a class="main-nav-link" href="/about">About</a></td>
                
                <td>
                    
    <div class="search-form">
        <input type="text" class="ins-search-input search-form-input" placeholder="Search" />
    </div>

                </td>
            </tr>
        </table>
    </div>
</header>

        <div class="outer">
            
                

<aside id="profile">
    <div class="inner profile-inner">
        <div class="base-info profile-block">
            <img id="avatar" src="https://raw.githubusercontent.com/xieguigang/xieguigang.github.io-hexo/master/images/my/WeChatImage636059681622385094.jpg" />
            <h2 id="name">xieguigang</h2>
            <h3 id="title">GCModeller chief developer</h3>
            <span id="location"><i class="fa fa-map-marker"></i>GuiLin, China (中國 · 桂林)</span>
            <a id="follow" target="_blank" href="https://github.com/xieguigang">FOLLOW</a>
        </div>
        <div class="article-info profile-block">
            <div class="article-info-block">
                55
                <span>posts</span>
            </div>
            <div class="article-info-block">
                112
                <span>tags</span>
            </div>
        </div>
        
        <div class="profile-block social-links">
            <table>
                <tr>
                    
                    
                    <td>
                        <a href="http://github.com/xieguigang" target="_blank" title="github" class=tooltip>
                            <i class="fa fa-github"></i>
                        </a>
                    </td>
                    
                    <td>
                        <a href="/" target="_blank" title="twitter" class=tooltip>
                            <i class="fa fa-twitter"></i>
                        </a>
                    </td>
                    
                    <td>
                        <a href="/" target="_blank" title="rss" class=tooltip>
                            <i class="fa fa-rss"></i>
                        </a>
                    </td>
                    
                </tr>
            </table>
        </div>
        
    </div>
</aside>

            
            <section id="main"><article id="post-fst" class="article article-type-post" itemscope itemprop="blogPost">
    <div class="article-inner">
        
        
        
            <header class="article-header">
                
    
        <h1 class="article-title" itemprop="name">
            Calculating F-statistics from genotypic data
        </h1>
    

                
                    <div class="article-meta">
                        
    <div class="article-date">
        <i class="fa fa-calendar"></i>
        <a href="/2016/08/03/fst/">
            <time datetime="2016-08-02T16:00:00.000Z" itemprop="datePublished">2016-08-03</time>
        </a>
    </div>


                        
                        
    <div class="article-tag">
        <i class="fa fa-tag"></i>
        <a class="tag-link" href="/tags/algorithm/">algorithm</a>, <a class="tag-link" href="/tags/genetics/">genetics</a>, <a class="tag-link" href="/tags/github/">github</a>, <a class="tag-link" href="/tags/vb-net/">vb.net</a>
    </div>

                    </div>
                
            </header>
        
        <div class="article-entry" itemprop="articleBody">
        
            
            <blockquote>
<p>CLI_tools source code on github: <a href="https://github.com/genetics-potato/pairwise.fst" target="_blank" rel="external">pairwise.fst</a></p>
</blockquote>
<h6 id="Genotype"><a href="#Genotype" class="headerlink" title="Genotype"></a>Genotype</h6><table>
<thead>
<tr>
<th>Populations</th>
<th>AA</th>
<th>Aa</th>
<th>aa</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subpopulation 1</td>
<td>125</td>
<td>250</td>
<td>125</td>
</tr>
<tr>
<td>Subpopulation 2</td>
<td>50</td>
<td>30</td>
<td>20</td>
</tr>
<tr>
<td>Subpopulation 3</td>
<td>100</td>
<td>500</td>
<td>400</td>
</tr>
</tbody>
</table>
<p><em>N</em> (number of individuals genotyped. The sum of each of the rows in the table above):</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Population 1:   500</span><br><span class="line">Population 2:   100</span><br><span class="line">Population 3:   1,000</span><br></pre></td></tr></table></figure>
<p>Remember that the number of alleles is <strong>TWICE</strong> the number of genotypes.</p>
<a id="more"></a>
<p><strong>Step 1.  Calculate the gene (allele) frequencies</strong></p>
<p>Each homozygote will have two alleles, each heterozygote will have one allele.  Note that the denominator will be twice <code>Ni</code> (twice as many alleles as individuals).</p>
<blockquote>
<p><strong>Eqns FST.1</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p1.png" alt=""></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p2.png" alt=""></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/p3.png" alt=""></p>
<ul>
<li><code>p1</code> means the frequency of allele <code>A</code> in population 1.</li>
</ul>
</blockquote>
<p><strong>Step 2.  Calculate the expected genotypic counts under Hardy-Weinberg Equilibrium</strong>, and then calculate the <strong>excess or deficiency of homozygotes in each subpopulation</strong>:</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">Pop. 1     Expected AA = 500*0.5^2      = 125     (= observed)</span><br><span class="line">           Expected Aa = 500*2*0.5*0.5  = 250     (= observed)</span><br><span class="line">           Expected aa = 500*0.5^2      = 125     (= observed)</span><br><span class="line"></span><br><span class="line">Pop. 2     Expected AA = 100*0.65^2        = 42.25   (observed has excess of 7.75)</span><br><span class="line">           Expected Aa = 100*2*0.65*0.35   =  45.5   (observed has deficit of 15.5)</span><br><span class="line">           Expected aa = 100*0.352         = 12.25   (observed has excess of 7.75)</span><br></pre></td></tr></table></figure>
<p>Note that sum of two types of homozygote excess = amount of heterozygote deficiency. These quantities have to balance (it’s a mathematical necessity, given that <code>p + q =1</code>.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Pop. 3     Expected AA = 1,000*0.35^2      = 122.5  (observed has deficiency of 22.5)</span><br><span class="line">           Expected Aa = 1,000*2*0.65*0.35 =  455   (observed has excess of 45)</span><br><span class="line">           Expected aa = 1,000*0.352       = 422.5  (observed has deficiency of 22.5)</span><br></pre></td></tr></table></figure>
<p>Summary of homozygote deficiency or excess relative to <code>HWE</code>:</p>
<ul>
<li>Pop. 1.  Observed = Expected: perfect fit</li>
<li>Pop. 2.  Excess of 15.5 homozygotes: some inbreeding</li>
<li>Pop. 3.  Deficiency of 45 homozygotes: outbred or experiencing a Wahlund effect (isolate breaking).</li>
</ul>
<p><strong>Step 3.  Calculate the local <em>observed</em> heterozygosities</strong> of each subpopulation (we will call them Hobs s, where the s  subscript refers to the <code>sth</code> of <code>n</code> populations  – 3 in this example).<br>Here we count genotypes:</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Hobs 1 = 250/500  = 0.5</span><br><span class="line">Hobs 2 =  30/100  = 0.3</span><br><span class="line">Hobs 3 = 500/1000 = 0.5</span><br></pre></td></tr></table></figure>
<p><strong>Step 4.  Calculate the local expected heterozygosity, or gene diversity, of each subpopulation</strong> (modified version of Eqn 35.1):</p>
<blockquote>
<p><strong>Eqns FST.2</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp1.png" alt=""><br><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp2.png" alt=""><br><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/H_exp3.png" alt=""></p>
<p>(With two alleles it would actually be easier to use <code>2pq</code> than to use the summation format of Eqn 33.1)</p>
</blockquote>
<p><strong>Notation</strong>: Note that I am using <code>p1</code> and <code>q1</code> here (where the subscripts refer to subpopulations 1 through 3). We would need to use multiple subscripts if we were using the notation of Eqn 35.1 where the alleles are <code>pi</code> (and the i refer to alleles 1 to k).  Indeed, with real multi-locus multipopulation data, we would have a triple summation and three subscripts; one for alleles (<code>i =1 to k</code>), one for the loci (l =1 tom), and one for subpopulations (s = 1 to n).</p>
<p><strong>Step 5.  Calculate the local inbreeding coefficient of each subpopulation</strong> (same as Eqn 35.4, except that we are subscripting for the subpopulations):</p>
<blockquote>
<p><strong>Eqn FST.3</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/Fs.png" alt=""></p>
<p>where <em>s</em> (<code>s = 1 to 3</code>) refers to the subpopulation</p>
</blockquote>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">F1 = (0.5 - 0.5) / 0.5     =  0</span><br><span class="line">F2 = (0.455 - 0.3) / 0.455 =  0.341 [positive F means fewer heterozygotes than expected indicates inbreeding]</span><br><span class="line">F3 = (0.455 — 0.5) / 0.455 = -0.099 [negative F means more heterozygotes than expected means excess outbreeding]</span><br></pre></td></tr></table></figure>
<p><strong>Step 6. Calculate <code>p</code> (p-bar, the frequency of allele A) over the total population.</strong><br>[Note that if we had more alleles we could put this and Step 7 all together as a single “global gene frequencies” step, or have one for each allele frequency].</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">(2*125+250+2*50+30+2*100+500)/(1000+200+2000)=0.4156  &#123;genotype splitting method&#125;</span><br></pre></td></tr></table></figure>
<p>or (yields same answer)<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">(0.5*1000+0.65*200+0.35*2000)/(1000+200+2000)=0.4156  &#123;using Eqn FST.1 values for ps&#125;</span><br></pre></td></tr></table></figure></p>
<p>Note that we weight by <strong>population size</strong></p>
<p><strong>Step 7.  Calculate <code>q</code> (q-bar, the frequency of allele a) over the total population</strong></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">(2*125+250+2*20+30+2*400+500)/(1000+200+2000)=0.5844</span><br></pre></td></tr></table></figure>
<p>Check: <code>p-bar + q-bar = 0.4156 + 0.5844 = 1.0</code> (as required by Eqn 31.1).<br>The check doesn’t guarantee that our result is correct, but if they don’t sum to one, we know we miscalculated.</p>
<p><strong>Step 8.  Calculate the global heterozygosity indices (over <em>I</em>ndividuals, <em>S</em>ubpopulations and <em>T</em>otal population)</strong><br>Note that the first two calculations employ a weighted average of the values in the whole set of subpopulations.</p>
<p><code>HI</code> based on <strong>observed</strong> heterozygosities in <strong>individuals</strong> in subpopulations</p>
<blockquote>
<p><strong>Eqn FST.4</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HI.png" alt=""></p>
</blockquote>
<p><code>HS</code> based on <strong>expected</strong> heterozygosities in <strong>subpopulations</strong></p>
<blockquote>
<p><strong>Eqn FST.5</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HS.png" alt=""></p>
</blockquote>
<p><code>HT</code> based on <strong>expected</strong> heterozygosities for overall total population (using global allele frequencies and a modified form of Eqn 35.1):</p>
<blockquote>
<p><strong>Eqn FST.6</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/HT.png" alt=""></p>
</blockquote>
<p>or we could calculate it as <code>2*p-bar *q-bar   = 2 * 0.4156 * 0.5844  = 0.4858</code></p>
<p><strong>Step 9.  CALCULATE THE GLOBAL <code>F-STATISTICS</code></strong><br>Compare and contrast the global FISbelow with the <strong>local inbreeding coefficient</strong> <code>Fs</code> of Step 5.<br>Here we are using a weighted average of the individual heterozygosities over all the subpopulations.<br>Both <code>FIS</code> and <code>Fs</code> are, however, based on the <code>observed</code> heterozygosities, whereas <code>FST</code> and <code>FIT</code> are based   on expected heterozygosities.</p>
<blockquote>
<p><strong>Eqn FST.7</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FIS.png" alt=""></p>
<p><strong>Eqn FST.8</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FST.png" alt=""></p>
<p><strong>Eqn FST.9</strong></p>
<p><img src="https://raw.githubusercontent.com/genetics-potato/pairwise.fst/master/images/FIT.png" alt=""></p>
</blockquote>
<p><strong>Notation note</strong>: the subscripts <code>I</code>, <code>S</code>, and <code>T</code> are not summation subscripts. They simply indicate the level of our analysis.  Likewise, the <code>s</code> on <code>Fs</code> in Step 5 or on the <code>ps</code> in Step 1 (the <code>s</code> was implicit there) just tell us what we are referring to. In contrast, the subscripts for <code>Eqn 35.1</code> and <code>35.2</code> are used in summations and change as we work through the pieces of the calculation.</p>
<p><strong>Step 10.  Finally, draw some conclusions about the genetic structure of the population and its subpopulations</strong>.</p>
<ul>
<li>One of the possible HWE conclusions we could make:<ul>
<li>Pop. 1 is consistent with <code>HWE</code> (results of Step 2)</li>
</ul>
</li>
<li>Two of the possible “local inbreeding” conclusions we could make from Step 5:<ul>
<li>Pop. 2 is inbred (results of Step 5), and</li>
<li>Pop. 3 may have disassortative mating or be experiencing a Wahlund effect (more heterozygotes than expected).</li>
</ul>
</li>
<li>Conclusion concerning overall degree of genetic differentiation (<code>FST</code>)<ul>
<li>Subdivision of populations, possibly due to genetic drift, accounts for approx. 3.4% of the total genetic variation (result of Eqn FST.8 FST calculation in Step 9),</li>
</ul>
</li>
<li>No excess or deficiency of heterozygotes over the total population (<code>FIT</code> is nearly zero).</li>
</ul>
<blockquote>
<p><img src="https://raw.githubusercontent.com/xieguigang/xieguigang.github.io-hexo/master/images/qrcode/Calculating_F-statistics_from_genotypic_data.png" alt="Population Ecology, ZOO 4400/5400, Spring 2013"></p>
</blockquote>

        
        </div>
        <footer class="article-footer">
            <div class="share-container">



</div>

    <a data-url="http://blog.xieguigang.me/2016/08/03/fst/" data-id="cire2riiv0001y8v46ro3pinw" class="article-share-link"><i class="fa fa-share"></i>Share</a>
<script>
    (function ($) {
        // Prevent duplicate binding
        if (typeof(__SHARE_BUTTON_BINDED__) === 'undefined' || !__SHARE_BUTTON_BINDED__) {
            __SHARE_BUTTON_BINDED__ = true;
        } else {
            return;
        }
        $('body').on('click', function() {
            $('.article-share-box.on').removeClass('on');
        }).on('click', '.article-share-link', function(e) {
            e.stopPropagation();

            var $this = $(this),
                url = $this.attr('data-url'),
                encodedUrl = encodeURIComponent(url),
                id = 'article-share-box-' + $this.attr('data-id'),
                offset = $this.offset(),
                box;

            if ($('#' + id).length) {
                box = $('#' + id);

                if (box.hasClass('on')){
                    box.removeClass('on');
                    return;
                }
            } else {
                var html = [
                    '<div id="' + id + '" class="article-share-box">',
                        '<input class="article-share-input" value="' + url + '">',
                        '<div class="article-share-links">',
                            '<a href="https://twitter.com/intent/tweet?url=' + encodedUrl + '" class="fa fa-twitter article-share-twitter" target="_blank" title="Twitter"></a>',
                            '<a href="https://www.facebook.com/sharer.php?u=' + encodedUrl + '" class="fa fa-facebook article-share-facebook" target="_blank" title="Facebook"></a>',
                            '<a href="http://pinterest.com/pin/create/button/?url=' + encodedUrl + '" class="fa fa-pinterest article-share-pinterest" target="_blank" title="Pinterest"></a>',
                            '<a href="https://plus.google.com/share?url=' + encodedUrl + '" class="fa fa-google article-share-google" target="_blank" title="Google+"></a>',
                        '</div>',
                    '</div>'
                ].join('');

              box = $(html);

              $('body').append(box);
            }

            $('.article-share-box.on').hide();

            box.css({
                top: offset.top + 25,
                left: offset.left
            }).addClass('on');

        }).on('click', '.article-share-box', function (e) {
            e.stopPropagation();
        }).on('click', '.article-share-box-input', function () {
            $(this).select();
        }).on('click', '.article-share-box-link', function (e) {
            e.preventDefault();
            e.stopPropagation();

            window.open(this.href, 'article-share-box-window-' + Date.now(), 'width=500,height=450');
        });
    })(jQuery);
</script>

            
    
        <a href="http://blog.xieguigang.me/2016/08/03/fst/#comments" class="article-comment-link disqus-comment-count" data-disqus-url="http://blog.xieguigang.me/2016/08/03/fst/">Comments</a>
    

        </footer>
    </div>
    
        
<nav id="article-nav">
    
        <a href="/2016/08/04/The_Screen_Behind_The_Mirror/" id="article-nav-newer" class="article-nav-link-wrap">
            <strong class="article-nav-caption">Newer</strong>
            <div class="article-nav-title">
                
                    The Screen Behind The Mirror
                
            </div>
        </a>
    
    
        <a href="/2016/08/02/MCMXC_A_D/" id="article-nav-older" class="article-nav-link-wrap">
            <strong class="article-nav-caption">Older</strong>
            <div class="article-nav-title">MCMXC, A.D.</div>
        </a>
    
</nav>


    
</article>


    
    <section id="comments">
    
        
    <div id="disqus_thread">
        <noscript>Please enable JavaScript to view the <a href="//disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    </div>

    
    </section>

</section>
            
                <aside id="sidebar">
   
        
    
        
    <div class="widget-wrap">
        <h3 class="widget-title">archives</h3>
        <div class="widget">
            <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2016/09/">September 2016</a><span class="archive-list-count">4</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2016/08/">August 2016</a><span class="archive-list-count">23</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2016/07/">July 2016</a><span class="archive-list-count">5</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2016/06/">June 2016</a><span class="archive-list-count">22</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2015/10/">October 2015</a><span class="archive-list-count">1</span></li></ul>
        </div>
    </div>

    
        
    <div class="widget-wrap">
        <h3 class="widget-title">tag cloud</h3>
        <div class="widget tagcloud">
            <a href="/tags/3D/" style="font-size: 10px;">3D</a> <a href="/tags/3d/" style="font-size: 10px;">3d</a> <a href="/tags/3d-graphics/" style="font-size: 10px;">3d graphics</a> <a href="/tags/Array/" style="font-size: 10px;">Array</a> <a href="/tags/BI/" style="font-size: 11.67px;">BI</a> <a href="/tags/CLI/" style="font-size: 10px;">CLI</a> <a href="/tags/Chinese/" style="font-size: 10px;">Chinese</a> <a href="/tags/Chinese-rural-dog/" style="font-size: 10px;">Chinese rural dog</a> <a href="/tags/CodeProject/" style="font-size: 15px;">CodeProject</a> <a href="/tags/Compiler/" style="font-size: 10px;">Compiler</a> <a href="/tags/Desktop/" style="font-size: 10px;">Desktop</a> <a href="/tags/Dictionary/" style="font-size: 10px;">Dictionary</a> <a href="/tags/Enigma/" style="font-size: 13.33px;">Enigma</a> <a href="/tags/GAM/" style="font-size: 10px;">GAM</a> <a href="/tags/GCModeller/" style="font-size: 11.67px;">GCModeller</a> <a href="/tags/HoloLens/" style="font-size: 10px;">HoloLens</a> <a href="/tags/IE/" style="font-size: 10px;">IE</a> <a href="/tags/Linq/" style="font-size: 10px;">Linq</a> <a href="/tags/Lumia/" style="font-size: 10px;">Lumia</a> <a href="/tags/MSDN/" style="font-size: 10px;">MSDN</a> <a href="/tags/Manhattan-Plot/" style="font-size: 10px;">Manhattan Plot</a> <a href="/tags/Markdown/" style="font-size: 10px;">Markdown</a> <a href="/tags/Microsoft/" style="font-size: 10px;">Microsoft</a> <a href="/tags/OMIM/" style="font-size: 10px;">OMIM</a> <a href="/tags/PDB/" style="font-size: 10px;">PDB</a> <a href="/tags/Phylostratigraphic/" style="font-size: 10px;">Phylostratigraphic</a> <a href="/tags/PowerBI/" style="font-size: 10px;">PowerBI</a> <a href="/tags/R/" style="font-size: 15px;">R</a> <a href="/tags/R-squared/" style="font-size: 10px;">R-squared</a> <a href="/tags/SDK-document/" style="font-size: 10px;">SDK document</a> <a href="/tags/SNP/" style="font-size: 15px;">SNP</a> <a href="/tags/SVG/" style="font-size: 10px;">SVG</a> <a href="/tags/String/" style="font-size: 10px;">String</a> <a href="/tags/Surface-Book/" style="font-size: 10px;">Surface Book</a> <a href="/tags/Syntax/" style="font-size: 10px;">Syntax</a> <a href="/tags/UI/" style="font-size: 10px;">UI</a> <a href="/tags/VisualStudio/" style="font-size: 10px;">VisualStudio</a> <a href="/tags/WPF/" style="font-size: 10px;">WPF</a> <a href="/tags/XAML/" style="font-size: 10px;">XAML</a> <a href="/tags/YAML/" style="font-size: 10px;">YAML</a> <a href="/tags/algorithm/" style="font-size: 10px;">algorithm</a> <a href="/tags/architecture/" style="font-size: 11.67px;">architecture</a> <a href="/tags/art-work/" style="font-size: 10px;">art work</a> <a href="/tags/async/" style="font-size: 10px;">async</a> <a href="/tags/bash/" style="font-size: 10px;">bash</a> <a href="/tags/check-point/" style="font-size: 10px;">check-point</a> <a href="/tags/chrome/" style="font-size: 10px;">chrome</a> <a href="/tags/circos/" style="font-size: 11.67px;">circos</a> <a href="/tags/cytoscape-js/" style="font-size: 11.67px;">cytoscape.js</a> <a href="/tags/d3js/" style="font-size: 13.33px;">d3js</a> <a href="/tags/data-analysis/" style="font-size: 10px;">data analysis</a> <a href="/tags/data-visualization/" style="font-size: 10px;">data visualization</a> <a href="/tags/desktop/" style="font-size: 10px;">desktop</a> <a href="/tags/document/" style="font-size: 10px;">document</a> <a href="/tags/elementary-os/" style="font-size: 10px;">elementary os</a> <a href="/tags/firefox/" style="font-size: 10px;">firefox</a> <a href="/tags/force-directed/" style="font-size: 10px;">force directed</a> <a href="/tags/gdi/" style="font-size: 11.67px;">gdi+</a> <a href="/tags/genetics/" style="font-size: 13.33px;">genetics</a> <a href="/tags/github/" style="font-size: 16.67px;">github</a> <a href="/tags/heatmap/" style="font-size: 10px;">heatmap</a> <a href="/tags/html5/" style="font-size: 10px;">html5</a> <a href="/tags/interface/" style="font-size: 10px;">interface</a> <a href="/tags/javascript/" style="font-size: 18.33px;">javascript</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/json/" style="font-size: 11.67px;">json</a> <a href="/tags/language/" style="font-size: 10px;">language</a> <a href="/tags/languages/" style="font-size: 10px;">languages</a> <a href="/tags/libzplay/" style="font-size: 10px;">libzplay</a> <a href="/tags/linux/" style="font-size: 10px;">linux</a> <a href="/tags/math/" style="font-size: 10px;">math</a> <a href="/tags/mgcv/" style="font-size: 10px;">mgcv</a> <a href="/tags/music/" style="font-size: 16.67px;">music</a> <a href="/tags/my-works/" style="font-size: 10px;">my works</a> <a href="/tags/mysql/" style="font-size: 10px;">mysql</a> <a href="/tags/nodejs/" style="font-size: 10px;">nodejs</a> <a href="/tags/note/" style="font-size: 18.33px;">note</a> <a href="/tags/open-source/" style="font-size: 10px;">open-source</a> <a href="/tags/perl/" style="font-size: 10px;">perl</a> <a href="/tags/pipeline/" style="font-size: 10px;">pipeline</a> <a href="/tags/re0/" style="font-size: 10px;">re0</a> <a href="/tags/rem/" style="font-size: 10px;">rem</a> <a href="/tags/repost/" style="font-size: 16.67px;">repost</a> <a href="/tags/schema/" style="font-size: 11.67px;">schema</a> <a href="/tags/statics/" style="font-size: 10px;">statics</a> <a href="/tags/svg/" style="font-size: 10px;">svg</a> <a href="/tags/threading/" style="font-size: 10px;">threading</a> <a href="/tags/threadpool/" style="font-size: 10px;">threadpool</a> <a href="/tags/tricks/" style="font-size: 10px;">tricks</a> <a href="/tags/ubuntu/" style="font-size: 10px;">ubuntu</a> <a href="/tags/user-agent/" style="font-size: 10px;">user-agent</a> <a href="/tags/vb-net/" style="font-size: 20px;">vb.net</a> <a href="/tags/web/" style="font-size: 10px;">web</a> <a href="/tags/web-api/" style="font-size: 10px;">web api</a> <a href="/tags/web-server/" style="font-size: 10px;">web server</a> <a href="/tags/win10/" style="font-size: 10px;">win10</a> <a href="/tags/works/" style="font-size: 10px;">works</a> <a href="/tags/中華田園犬/" style="font-size: 10px;">中華田園犬</a> <a href="/tags/前端/" style="font-size: 10px;">前端</a> <a href="/tags/前端设计/" style="font-size: 13.33px;">前端设计</a> <a href="/tags/数据分析/" style="font-size: 10px;">数据分析</a> <a href="/tags/数据可视化/" style="font-size: 11.67px;">数据可视化</a> <a href="/tags/数据视图/" style="font-size: 10px;">数据视图</a> <a href="/tags/架构/" style="font-size: 11.67px;">架构</a> <a href="/tags/架构设计/" style="font-size: 10px;">架构设计</a> <a href="/tags/线程池/" style="font-size: 10px;">线程池</a> <a href="/tags/转载/" style="font-size: 10px;">转载</a> <a href="/tags/转载搬运/" style="font-size: 16.67px;">转载搬运</a> <a href="/tags/连锁不平衡/" style="font-size: 11.67px;">连锁不平衡</a> <a href="/tags/遗传学/" style="font-size: 11.67px;">遗传学</a> <a href="/tags/非线性拟合/" style="font-size: 10px;">非线性拟合</a> <a href="/tags/鹿乃/" style="font-size: 10px;">鹿乃</a>
        </div>
    </div>

    
        
    <div class="widget-wrap widget-list">
        <h3 class="widget-title">links</h3>
        <div class="widget">
            <ul>
                
                    <li>
                        <a href="http://framework-docs.xieguigang.me/">VisualBasic Framework</a>
                    </li>
                
                    <li>
                        <a href="http://hexo.io">Hexo</a>
                    </li>
                
                    <li>
                        <a href="http://gcmodeller.org">http://gcmodeller.org</a>
                    </li>
                
                    <li>
                        <a href="https://github.com/genetics-potato">https://github.com/genetics-potato</a>
                    </li>
                
            </ul>
        </div>
    </div>


    
    <div id="toTop" class="fa fa-angle-up"></div>
</aside>
            
        </div>
        <footer id="footer">
    <div class="outer">
        <div id="footer-info" class="inner">
            &copy; 2016 しゃけい よしつな<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>. Theme by <a href="http://github.com/ppoffice">PPOffice</a>
        </div>
    </div>
</footer>
        
    
    <script>
    var disqus_config = function () {
        
            this.page.url = 'http://blog.xieguigang.me/2016/08/03/fst/';
        
        this.page.identifier = 'fst';
    };
    (function() { 
        var d = document, s = d.createElement('script');  
        s.src = '//' + 'xieguigang' + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>



    
        <script src="/vendor/fancybox/jquery.fancybox.pack.js"></script>
    


<!-- Custom Scripts -->
<script src="/js/main.js"></script>

    </div>
</body>
</html>